<!DOCTYPE html>

<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <meta name="mobile-web-app-capable" content="yes">
    <title>
        ChIP-seq2 - HackMD
    </title>
    <link rel="icon" type="image/png" href="https://hackmd.io/favicon.png">
    <link rel="apple-touch-icon" href="https://hackmd.io/apple-touch-icon.png">

    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/ionicons/2.0.1/css/ionicons.min.css" integrity="sha256-3iu9jgsy9TpTwXKb7bNQzqWekRX7pPK+2OLj3R922fo=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/octicons/3.5.0/octicons.min.css" integrity="sha256-QiWfLIsCT02Sdwkogf6YMiQlj4NE84MKkzEMkZnMGdg=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.5.1/themes/prism.min.css" integrity="sha256-vtR0hSWRc3Tb26iuN2oZHt3KRUomwTufNIf5/4oeCyg=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@hackmd/emojify.js@2.1.0/dist/css/basic/emojify.min.css" integrity="sha256-UOrvMOsSDSrW6szVLe8ZDZezBxh5IoIfgTwdNDgTjiU=" crossorigin="anonymous" />
    <style>
        @import url(https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,500,500i|Source+Code+Pro:300,400,500|Source+Sans+Pro:300,300i,400,400i,600,600i|Source+Serif+Pro&subset=latin-ext);.hljs{background:#fff;color:#333;display:block;overflow-x:auto;padding:.5em}.hljs-comment,.hljs-meta{color:#969896}.hljs-emphasis,.hljs-quote,.hljs-string,.hljs-strong,.hljs-template-variable,.hljs-variable{color:#df5000}.hljs-keyword,.hljs-selector-tag,.hljs-type{color:#a71d5d}.hljs-attribute,.hljs-bullet,.hljs-literal,.hljs-number,.hljs-symbol{color:#0086b3}.hljs-built_in,.hljs-builtin-name{color:#005cc5}.hljs-name,.hljs-section{color:#63a35c}.hljs-tag{color:#333}.hljs-attr,.hljs-selector-attr,.hljs-selector-class,.hljs-selector-id,.hljs-selector-pseudo,.hljs-title{color:#795da3}.hljs-addition{background-color:#eaffea;color:#55a532}.hljs-deletion{background-color:#ffecec;color:#bd2c00}.hljs-link{text-decoration:underline}.markdown-body{word-wrap:break-word;font-size:16px;line-height:1.5}.markdown-body:after,.markdown-body:before{content:"";display:table}.markdown-body:after{clear:both}.markdown-body>:first-child{margin-top:0!important}.markdown-body>:last-child{margin-bottom:0!important}.markdown-body a:not([href]){color:inherit;text-decoration:none}.markdown-body .absent{color:#c00}.markdown-body .anchor{float:left;line-height:1;margin-left:-20px;padding-right:4px}.markdown-body .anchor:focus{outline:none}.markdown-body blockquote,.markdown-body dl,.markdown-body ol,.markdown-body p,.markdown-body pre,.markdown-body table,.markdown-body ul{margin-bottom:16px;margin-top:0}.markdown-body hr{background-color:#e7e7e7;border:0;height:.25em;margin:24px 0;padding:0}.markdown-body blockquote{border-left:.25em solid #ddd;color:#777;font-size:16px;padding:0 1em}.markdown-body blockquote>:first-child{margin-top:0}.markdown-body blockquote>:last-child{margin-bottom:0}.markdown-body kbd,.popover kbd{background-color:#fcfcfc;border:1px solid;border-color:#ccc #ccc #bbb;border-radius:3px;box-shadow:inset 0 -1px 0 #bbb;color:#555;display:inline-block;font-size:11px;line-height:10px;padding:3px 5px;vertical-align:middle}.markdown-body .loweralpha{list-style-type:lower-alpha}.markdown-body h1,.markdown-body h2,.markdown-body h3,.markdown-body h4,.markdown-body h5,.markdown-body h6{font-weight:600;line-height:1.25;margin-bottom:16px;margin-top:24px}.markdown-body h1 .octicon-link,.markdown-body h2 .octicon-link,.markdown-body h3 .octicon-link,.markdown-body h4 .octicon-link,.markdown-body h5 .octicon-link,.markdown-body h6 .octicon-link{color:#000;vertical-align:middle;visibility:hidden}.markdown-body h1:hover .anchor,.markdown-body h2:hover .anchor,.markdown-body h3:hover .anchor,.markdown-body h4:hover .anchor,.markdown-body h5:hover .anchor,.markdown-body h6:hover .anchor{text-decoration:none}.markdown-body h1:hover .anchor .octicon-link,.markdown-body h2:hover .anchor .octicon-link,.markdown-body h3:hover .anchor .octicon-link,.markdown-body h4:hover .anchor .octicon-link,.markdown-body h5:hover .anchor .octicon-link,.markdown-body h6:hover .anchor .octicon-link{visibility:visible}.markdown-body h1 code,.markdown-body h1 tt,.markdown-body h2 code,.markdown-body h2 tt,.markdown-body h3 code,.markdown-body h3 tt,.markdown-body h4 code,.markdown-body h4 tt,.markdown-body h5 code,.markdown-body h5 tt,.markdown-body h6 code,.markdown-body h6 tt{font-size:inherit}.markdown-body h1{font-size:2em}.markdown-body h1,.markdown-body h2{border-bottom:1px solid #eee;padding-bottom:.3em}.markdown-body h2{font-size:1.5em}.markdown-body h3{font-size:1.25em}.markdown-body h4{font-size:1em}.markdown-body h5{font-size:.875em}.markdown-body h6{color:#777;font-size:.85em}.markdown-body ol,.markdown-body ul{padding-left:2em}.markdown-body ol.no-list,.markdown-body ul.no-list{list-style-type:none;padding:0}.markdown-body ol ol,.markdown-body ol ul,.markdown-body ul ol,.markdown-body ul ul{margin-bottom:0;margin-top:0}.markdown-body li>p{margin-top:16px}.markdown-body li+li{padding-top:.25em}.markdown-body dl{padding:0}.markdown-body dl dt{font-size:1em;font-style:italic;font-weight:700;margin-top:16px;padding:0}.markdown-body dl dd{margin-bottom:16px;padding:0 16px}.markdown-body table{display:block;overflow:auto;width:100%;word-break:normal;word-break:keep-all}.markdown-body table th{font-weight:700}.markdown-body table td,.markdown-body table th{border:1px solid #ddd;padding:6px 13px}.markdown-body table tr{background-color:#fff;border-top:1px solid #ccc}.markdown-body table tr:nth-child(2n){background-color:#f8f8f8}.markdown-body img{background-color:#fff;box-sizing:initial;max-width:100%}.markdown-body img[align=right]{padding-left:20px}.markdown-body img[align=left]{padding-right:20px}.markdown-body .emoji{background-color:initial;max-width:none;vertical-align:text-top}.markdown-body span.frame{display:block;overflow:hidden}.markdown-body span.frame>span{border:1px solid #ddd;display:block;float:left;margin:13px 0 0;overflow:hidden;padding:7px;width:auto}.markdown-body span.frame span img{display:block;float:left}.markdown-body span.frame span span{clear:both;color:#333;display:block;padding:5px 0 0}.markdown-body span.align-center{clear:both;display:block;overflow:hidden}.markdown-body span.align-center>span{display:block;margin:13px auto 0;overflow:hidden;text-align:center}.markdown-body span.align-center span img{margin:0 auto;text-align:center}.markdown-body span.align-right{clear:both;display:block;overflow:hidden}.markdown-body span.align-right>span{display:block;margin:13px 0 0;overflow:hidden;text-align:right}.markdown-body span.align-right span img{margin:0;text-align:right}.markdown-body span.float-left{display:block;float:left;margin-right:13px;overflow:hidden}.markdown-body span.float-left span{margin:13px 0 0}.markdown-body span.float-right{display:block;float:right;margin-left:13px;overflow:hidden}.markdown-body span.float-right>span{display:block;margin:13px auto 0;overflow:hidden;text-align:right}.markdown-body code,.markdown-body tt{background-color:#0000000a;border-radius:3px;font-size:85%;margin:0;padding:.2em 0}.markdown-body code:after,.markdown-body code:before,.markdown-body tt:after,.markdown-body tt:before{content:"\00a0";letter-spacing:-.2em}.markdown-body code br,.markdown-body tt br{display:none}.markdown-body del code{text-decoration:inherit}.markdown-body pre{word-wrap:normal}.markdown-body pre>code{background:#0000;border:0;font-size:100%;margin:0;padding:0;white-space:pre;word-break:normal}.markdown-body .highlight{margin-bottom:16px}.markdown-body .highlight pre{margin-bottom:0;word-break:normal}.markdown-body .highlight pre,.markdown-body pre{background-color:#f7f7f7;border-radius:3px;font-size:85%;line-height:1.45;overflow:auto;padding:16px}.markdown-body pre code,.markdown-body pre tt{word-wrap:normal;background-color:initial;border:0;display:inline;line-height:inherit;margin:0;max-width:auto;overflow:visible;padding:0}.markdown-body pre code:after,.markdown-body pre code:before,.markdown-body pre tt:after,.markdown-body pre tt:before{content:normal}.markdown-body .csv-data td,.markdown-body .csv-data th{font-size:12px;line-height:1;overflow:hidden;padding:5px;text-align:left;white-space:nowrap}.markdown-body .csv-data .blob-line-num{background:#fff;border:0;padding:10px 8px 9px;text-align:right}.markdown-body .csv-data tr{border-top:0}.markdown-body .csv-data th{background:#f8f8f8;border-top:0;font-weight:700}.news .alert .markdown-body blockquote{border:0;padding:0 0 0 40px}.activity-tab .news .alert .commits,.activity-tab .news .markdown-body blockquote{padding-left:0}.task-list-item{list-style-type:none}.task-list-item label{font-weight:400}.task-list-item.enabled label{cursor:pointer}.task-list-item+.task-list-item{margin-top:3px}.task-list-item-checkbox{cursor:default!important;float:left;margin:.31em 0 .2em -1.3em!important;vertical-align:middle}.markdown-body{max-width:758px;overflow:visible!important;padding-bottom:40px;padding-top:40px;position:relative}.markdown-body .emoji{vertical-align:top}.markdown-body pre{border:inherit!important}.markdown-body code{color:inherit!important}.markdown-body pre code .wrapper{display:-moz-inline-flex;display:-ms-inline-flex;display:-o-inline-flex;display:inline-flex}.markdown-body pre code .gutter{float:left;overflow:hidden;-webkit-user-select:none;user-select:none}.markdown-body pre code .gutter.linenumber{border-right:3px solid #6ce26c!important;box-sizing:initial;color:#afafaf!important;cursor:default;display:inline-block;min-width:20px;padding:0 8px 0 0;position:relative;text-align:right;z-index:4}.markdown-body pre code .gutter.linenumber>span:before{content:attr(data-linenumber)}.markdown-body pre code .code{float:left;margin:0 0 0 16px}.markdown-body .gist .line-numbers{border-bottom:none;border-left:none;border-top:none}.markdown-body .gist .line-data{border:none}.markdown-body .gist table{border-collapse:inherit!important;border-spacing:0}.markdown-body code[data-gist-id]{background:none;padding:0}.markdown-body code[data-gist-id]:after,.markdown-body code[data-gist-id]:before{content:""}.markdown-body code[data-gist-id] .blob-num{border:unset}.markdown-body code[data-gist-id] table{margin-bottom:unset;overflow:unset}.markdown-body code[data-gist-id] table tr{background:unset}.markdown-body[dir=rtl] pre{direction:ltr}.markdown-body[dir=rtl] code{direction:ltr;unicode-bidi:embed}.markdown-body .alert>p:last-child{margin-bottom:0}.markdown-body pre.abc,.markdown-body pre.flow-chart,.markdown-body pre.graphviz,.markdown-body pre.mermaid,.markdown-body pre.sequence-diagram,.markdown-body pre.vega{background-color:inherit;border-radius:0;overflow:visible;text-align:center;white-space:inherit}.markdown-body pre.abc>code,.markdown-body pre.flow-chart>code,.markdown-body pre.graphviz>code,.markdown-body pre.mermaid>code,.markdown-body pre.sequence-diagram>code,.markdown-body pre.vega>code{text-align:left}.markdown-body pre.abc>svg,.markdown-body pre.flow-chart>svg,.markdown-body pre.graphviz>svg,.markdown-body pre.mermaid>svg,.markdown-body pre.sequence-diagram>svg,.markdown-body pre.vega>svg{height:100%;max-width:100%}.markdown-body pre>code.wrap{word-wrap:break-word;white-space:pre-wrap;white-space:-moz-pre-wrap;white-space:-pre-wrap;white-space:-o-pre-wrap}.markdown-body .alert>p:last-child,.markdown-body .alert>ul:last-child{margin-bottom:0}.markdown-body summary{display:list-item}.markdown-body summary:focus{outline:none}.markdown-body details summary{cursor:pointer}.markdown-body details:not([open])>:not(summary){display:none}.markdown-body figure{margin:1em 40px}.markdown-body .mark,.markdown-body mark{background-color:#fff1a7}.vimeo,.youtube{background-color:#000;background-position:50%;background-repeat:no-repeat;background-size:contain;cursor:pointer;display:table;overflow:hidden;text-align:center}.vimeo,.youtube{position:relative;width:100%}.youtube{padding-bottom:56.25%}.vimeo img{object-fit:contain;width:100%;z-index:0}.youtube img{object-fit:cover;z-index:0}.vimeo iframe,.youtube iframe,.youtube img{height:100%;left:0;position:absolute;top:0;width:100%}.vimeo iframe,.youtube iframe{vertical-align:middle;z-index:1}.vimeo .icon,.youtube .icon{color:#fff;height:auto;left:50%;opacity:.3;position:absolute;top:50%;transform:translate(-50%,-50%);transition:opacity .2s;width:auto;z-index:0}.vimeo:hover .icon,.youtube:hover .icon{opacity:.6;transition:opacity .2s}.slideshare .inner,.speakerdeck .inner{position:relative;width:100%}.slideshare .inner iframe,.speakerdeck .inner iframe{bottom:0;height:100%;left:0;position:absolute;right:0;top:0;width:100%}.figma{display:table;padding-bottom:56.25%;position:relative;width:100%}.figma iframe{border:1px solid #eee;bottom:0;height:100%;left:0;position:absolute;right:0;top:0;width:100%}.markmap-container{height:300px}.markmap-container>svg{height:100%;width:100%}.MJX_Assistive_MathML{display:none}#MathJax_Message{z-index:1000!important}.ui-infobar{color:#777;margin:25px auto -25px;max-width:760px;position:relative;z-index:2}.toc .invisable-node{list-style-type:none}.ui-toc{bottom:20px;position:fixed;z-index:998}.ui-toc.both-mode{margin-left:8px}.ui-toc.both-mode .ui-toc-label{border-bottom-left-radius:0;border-top-left-radius:0;height:40px;padding:10px 4px}.ui-toc-label{background-color:#e6e6e6;border:none;color:#868686;transition:opacity .2s}.ui-toc .open .ui-toc-label{color:#fff;opacity:1;transition:opacity .2s}.ui-toc-label:focus{background-color:#ccc;color:#000;opacity:.3}.ui-toc-label:hover{background-color:#ccc;opacity:1;transition:opacity .2s}.ui-toc-dropdown{margin-bottom:20px;margin-top:20px;max-height:70vh;max-width:45vw;overflow:auto;padding-left:10px;padding-right:10px;text-align:inherit;width:25vw}.ui-toc-dropdown>.toc{max-height:calc(70vh - 100px);overflow:auto}.ui-toc-dropdown[dir=rtl] .nav{letter-spacing:.0029em;padding-right:0}.ui-toc-dropdown a{overflow:hidden;text-overflow:ellipsis;white-space:pre}.ui-toc-dropdown .nav>li>a{color:#767676;display:block;font-size:13px;font-weight:500;padding:4px 20px}.ui-toc-dropdown .nav>li:first-child:last-child>ul,.ui-toc-dropdown .toc.expand ul{display:block}.ui-toc-dropdown .nav>li>a:focus,.ui-toc-dropdown .nav>li>a:hover{background-color:initial;border-left:1px solid #000;color:#000;padding-left:19px;text-decoration:none}.ui-toc-dropdown[dir=rtl] .nav>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav>li>a:hover{border-left:none;border-right:1px solid #000;padding-right:19px}.ui-toc-dropdown .nav>.active:focus>a,.ui-toc-dropdown .nav>.active:hover>a,.ui-toc-dropdown .nav>.active>a{background-color:initial;border-left:2px solid #000;color:#000;font-weight:700;padding-left:18px}.ui-toc-dropdown[dir=rtl] .nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav>.active>a{border-left:none;border-right:2px solid #000;padding-right:18px}.ui-toc-dropdown .nav .nav{display:none;padding-bottom:10px}.ui-toc-dropdown .nav>.active>ul{display:block}.ui-toc-dropdown .nav .nav>li>a{font-size:12px;font-weight:400;padding-bottom:1px;padding-left:30px;padding-top:1px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>a{padding-right:30px}.ui-toc-dropdown .nav .nav>li>ul>li>a{font-size:12px;font-weight:400;padding-bottom:1px;padding-left:40px;padding-top:1px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a{padding-right:40px}.ui-toc-dropdown .nav .nav>li>a:focus,.ui-toc-dropdown .nav .nav>li>a:hover{padding-left:29px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav .nav>li>a:hover{padding-right:29px}.ui-toc-dropdown .nav .nav>li>ul>li>a:focus,.ui-toc-dropdown .nav .nav>li>ul>li>a:hover{padding-left:39px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a:hover{padding-right:39px}.ui-toc-dropdown .nav .nav>.active:focus>a,.ui-toc-dropdown .nav .nav>.active:hover>a,.ui-toc-dropdown .nav .nav>.active>a{font-weight:500;padding-left:28px}.ui-toc-dropdown[dir=rtl] .nav .nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>a{padding-right:28px}.ui-toc-dropdown .nav .nav>.active>.nav>.active:focus>a,.ui-toc-dropdown .nav .nav>.active>.nav>.active:hover>a,.ui-toc-dropdown .nav .nav>.active>.nav>.active>a{font-weight:500;padding-left:38px}.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active>a{padding-right:38px}.markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang^=ja] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang=zh-tw] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang=zh-cn] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang^=ja]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang=zh-tw]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang=zh-cn]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang^=ja] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Meiryo UI,MS PGothic,ＭＳ Ｐゴシック,sans-serif}html[lang=zh-tw] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Microsoft JhengHei UI,微軟正黑UI,sans-serif}html[lang=zh-cn] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Microsoft YaHei UI,微软雅黑UI,sans-serif}html .ui-toc-dropdown[lang^=ja]{font-family:Source Sans Pro,Helvetica,Arial,Meiryo UI,MS PGothic,ＭＳ Ｐゴシック,sans-serif}html .ui-toc-dropdown[lang=zh-tw]{font-family:Source Sans Pro,Helvetica,Arial,Microsoft JhengHei UI,微軟正黑UI,sans-serif}html .ui-toc-dropdown[lang=zh-cn]{font-family:Source Sans Pro,Helvetica,Arial,Microsoft YaHei UI,微软雅黑UI,sans-serif}.ui-affix-toc{max-height:70vh;max-width:15vw;overflow:auto;position:fixed;top:0}.back-to-top,.expand-toggle,.go-to-bottom{color:#999;display:block;font-size:12px;font-weight:500;margin-left:10px;margin-top:10px;padding:4px 10px}.back-to-top:focus,.back-to-top:hover,.expand-toggle:focus,.expand-toggle:hover,.go-to-bottom:focus,.go-to-bottom:hover{color:#563d7c;text-decoration:none}.back-to-top,.go-to-bottom{margin-top:0}.ui-user-icon{background-position:50%;background-repeat:no-repeat;background-size:cover;border-radius:50%;display:block;height:20px;margin-bottom:2px;margin-right:5px;margin-top:2px;width:20px}.ui-user-icon.small{display:inline-block;height:18px;margin:0 0 .2em;vertical-align:middle;width:18px}.ui-infobar>small>span{line-height:22px}.ui-infobar>small .dropdown{display:inline-block}.ui-infobar>small .dropdown a:focus,.ui-infobar>small .dropdown a:hover{text-decoration:none}.ui-more-info{color:#888;cursor:pointer;vertical-align:middle}.ui-more-info .fa{font-size:16px}.ui-connectedGithub,.ui-published-note{color:#888}.ui-connectedGithub{line-height:23px;white-space:nowrap}.ui-connectedGithub a.file-path{color:#888;padding-left:22px;text-decoration:none}.ui-connectedGithub a.file-path:active,.ui-connectedGithub a.file-path:hover{color:#888;text-decoration:underline}.ui-connectedGithub .fa{font-size:20px}.ui-published-note .fa{font-size:20px;vertical-align:top}.unselectable{-webkit-user-select:none;-o-user-select:none;user-select:none}.selectable{-webkit-user-select:text;-o-user-select:text;user-select:text}.inline-spoiler-section{cursor:pointer}.inline-spoiler-section .spoiler-text{background-color:#333;border-radius:2px}.inline-spoiler-section .spoiler-text>*{opacity:0}.inline-spoiler-section .spoiler-img{filter:blur(10px)}.inline-spoiler-section.raw{background-color:#333;border-radius:2px}.inline-spoiler-section.raw>*{opacity:0}.inline-spoiler-section.unveil{cursor:auto}.inline-spoiler-section.unveil .spoiler-text{background-color:#3333331a}.inline-spoiler-section.unveil .spoiler-text>*{opacity:1}.inline-spoiler-section.unveil .spoiler-img{filter:none}@media print{blockquote,div,img,pre,table{page-break-inside:avoid!important}a[href]:after{font-size:12px!important}}.markdown-body.slides{color:#222;position:relative;z-index:1}.markdown-body.slides:before{background-color:currentColor;bottom:0;box-shadow:0 0 0 50vw;content:"";display:block;left:0;position:absolute;right:0;top:0;z-index:-1}.markdown-body.slides section[data-markdown]{background-color:#fff;margin-bottom:1.5em;position:relative;text-align:center}.markdown-body.slides section[data-markdown] code{text-align:left}.markdown-body.slides section[data-markdown]:before{content:"";display:block;padding-bottom:56.23%}.markdown-body.slides section[data-markdown]>div:first-child{left:1em;max-height:100%;overflow:hidden;position:absolute;right:1em;top:50%;transform:translateY(-50%)}.markdown-body.slides section[data-markdown]>ul{display:inline-block}.markdown-body.slides>section>section+section:after{border:3px solid #777;content:"";height:1.5em;position:absolute;right:1em;top:-1.5em}.site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,sans-serif}html[lang^=ja] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif}html[lang=zh-tw] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif}html[lang=zh-cn] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif}body{font-smoothing:subpixel-antialiased!important;-webkit-font-smoothing:subpixel-antialiased!important;-moz-osx-font-smoothing:auto!important;-webkit-overflow-scrolling:touch;font-family:Source Sans Pro,Helvetica,Arial,sans-serif;letter-spacing:.025em}html[lang^=ja] body{font-family:Source Sans Pro,Helvetica,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif}html[lang=zh-tw] body{font-family:Source Sans Pro,Helvetica,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif}html[lang=zh-cn] body{font-family:Source Sans Pro,Helvetica,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif}abbr[title]{border-bottom:none;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}abbr[data-original-title],abbr[title]{cursor:help}body.modal-open{overflow-y:auto;padding-right:0!important}svg{text-shadow:none}
    </style>
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
    	<script src="https://cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv.min.js" integrity="sha256-3Jy/GbSLrg0o9y5Z5n1uw0qxZECH7C6OQpVBgNFYa0g=" crossorigin="anonymous"></script>
    	<script src="https://cdnjs.cloudflare.com/ajax/libs/respond.js/1.4.2/respond.min.js" integrity="sha256-g6iAfvZp+nDQ2TdTR/VVKJf3bGro4ub5fvWSWVRi2NE=" crossorigin="anonymous"></script>
		<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js" integrity="sha256-8E4Is26QH0bD52WoQpcB+R/tcWQtpzlCojrybUd7Mxo=" crossorigin="anonymous"></script>
    <![endif]-->
</head>

<body>
    <div id="doc" class="markdown-body container-fluid comment-inner comment-enabled" data-hard-breaks="true"><h1 id="The-environment---task-sheet" data-id="The-environment---task-sheet"><a class="anchor hidden-xs" href="#The-environment---task-sheet" title="The-environment---task-sheet"><span class="octicon octicon-link"></span></a><span>The environment - task sheet</span></h1><h2 id="2-Get-to-know-the-system" data-id="2-Get-to-know-the-system"><a class="anchor hidden-xs" href="#2-Get-to-know-the-system" title="2-Get-to-know-the-system"><span class="octicon octicon-link"></span></a><span>2 Get to know the system</span></h2><ol>
<li><span>Run sinfo -lN on the master node.</span></li>
</ol><pre><code class="{bash} hljs"><span class="hljs-attribute">sinfo</span> -lN
</code></pre><p><span>How many nodes do you see? - 4</span><br>
<span>How many CPUs? - 12</span><br>
<span>How much RAM does each node have? - 121000</span></p><p><img src="https://i.imgur.com/K3KSP91.png" alt="" loading="lazy"></p><ol start="2">
<li><span>Run df -h /vol/COMPEPIWS/.</span></li>
</ol><pre><code class="{bash} hljs">df <span class="hljs-operator">-</span>h <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>.
</code></pre><p><span>What is the volume’s size? - 3.9T</span><br>
<img src="https://i.imgur.com/0pcn2Kg.png" alt="" loading="lazy"></p><ol start="3">
<li><span>Check the temp folder location by</span></li>
</ol><pre><code class="{bash} hljs"><span class="hljs-built_in">echo</span> <span class="hljs-variable">$TMPDIR</span>: /vol/COMPEPIWS/tmp
</code></pre><ol start="4">
<li><span>Your group working directory is</span></li>
</ol><pre><code>/vol/COMPEPIWS/groups/chipseq2
</code></pre><ol start="6">
<li><span>The raw data (fastq files) that you are going to use to run the pipelines is available under this folder:</span></li>
</ol><pre><code>/vol/COMPEPIWS/data/reduced/&lt;assay&gt;
</code></pre><p><img src="https://i.imgur.com/uv6MHkW.png" alt="" loading="lazy"></p><h2 id="3-Tips-and-Tricks" data-id="3-Tips-and-Tricks"><a class="anchor hidden-xs" href="#3-Tips-and-Tricks" title="3-Tips-and-Tricks"><span class="octicon octicon-link"></span></a><span>3 Tips and Tricks</span></h2><pre><code class="{bash} hljs"><span class="hljs-number">1.</span> $ ssh <span class="hljs-operator">-</span>X <span class="hljs-operator">-</span>A <span class="hljs-operator">-</span>i OpenSSH_private_key [ELIXIR_LOGIN]@<span class="hljs-number">193.175</span><span class="hljs-number">.249</span><span class="hljs-number">.28</span>
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">2.</span> $ ssh <span class="hljs-operator">-</span>X <span class="hljs-operator">-</span>A  <span class="hljs-operator">-</span>i OpenSSH_private_key bibigrid<span class="hljs-operator">-</span>worker<span class="hljs-number">-1</span><span class="hljs-number">-2</span><span class="hljs-operator">-</span>etqcgckjrdsvsxe
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">3</span>. $ <span class="hljs-keyword">screen </span>-S <span class="hljs-keyword">screen_1
</span></code></pre><pre><code class="{bash} hljs"><span class="hljs-number">4</span>. Ctrl-<span class="hljs-selector-tag">a</span> c
   Ctrl-<span class="hljs-selector-tag">a</span> c
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">5</span>. Ctrl-<span class="hljs-selector-tag">a</span> n
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">6</span>. Ctrl-<span class="hljs-selector-tag">a</span> d
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">7.</span> Ctrl<span class="hljs-operator">-</span>d
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">8.</span> Ctrl<span class="hljs-operator">-</span>d
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">9.</span> $ ssh <span class="hljs-operator">-</span>X <span class="hljs-operator">-</span>A <span class="hljs-operator">-</span>i OpenSSH_private_key [ELIXIR_LOGIN]@<span class="hljs-number">193.175</span><span class="hljs-number">.249</span><span class="hljs-number">.28</span>
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">10.</span> $ ssh <span class="hljs-operator">-</span>X <span class="hljs-operator">-</span>A  <span class="hljs-operator">-</span>i OpenSSH_private_key bibigrid<span class="hljs-operator">-</span>worker<span class="hljs-number">-1</span><span class="hljs-number">-2</span><span class="hljs-operator">-</span>etqcgckjrdsvsxe 
</code></pre><pre><code class="{bash} hljs">11. $ screen -<span class="hljs-built_in">ls</span>
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">12</span>. <span class="hljs-variable">$ </span>screen -r
</code></pre><pre><code class="{bash} hljs"><span class="hljs-number">13</span>. Ctrl-<span class="hljs-selector-tag">a</span> n
</code></pre><h2 id="4-Working-with-tables-on-the-command-line-or-awk" data-id="4-Working-with-tables-on-the-command-line-or-awk"><a class="anchor hidden-xs" href="#4-Working-with-tables-on-the-command-line-or-awk" title="4-Working-with-tables-on-the-command-line-or-awk"><span class="octicon octicon-link"></span></a><span>4 Working with tables on the command line, or awk</span></h2><ol start="3">
<li><span>Number of lines in this file: 671462</span></li>
</ol><pre><code class="{bash} hljs">wc <span class="hljs-operator">-</span>l <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>pipelines<span class="hljs-operator">/</span>references<span class="hljs-operator">/</span>genome_genes.gtf
</code></pre><ol start="4">
<li><span>Number of “exon” in this file (hint: grep, awk): 324748</span></li>
</ol><pre><code class="{bash} hljs">grep <span class="hljs-operator">-</span>o <span class="hljs-operator">-</span>i exon <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>pipelines<span class="hljs-operator">/</span>refrences<span class="hljs-operator">/</span>genome_genes.gtf <span class="hljs-operator">|</span> wc <span class="hljs-operator">-</span>l
</code></pre><ol start="5">
<li><span>Number of “exon” that are longer than 1000bp considering cloumns 4 and 5 as coordinates of the exon :19593</span></li>
</ol><pre><code class="{bash} hljs">awk <span class="hljs-string">'$3 = "exon" &amp;&amp; $5-$4 &gt; 1000'</span> <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>pipelines<span class="hljs-operator">/</span>refrences<span class="hljs-operator">/</span>genome_genes.gtf <span class="hljs-operator">&gt;</span> exon_genes.gtf
</code></pre><pre><code class="{bash} hljs">wc <span class="hljs-operator">-</span>l exon_genes.gtf <span class="hljs-number">19593</span> exon_genes.gtf
</code></pre><p><img src="https://i.imgur.com/QVdneN2.png" alt="" loading="lazy"></p><ol start="6">
<li><span>Number of “exon” that has Sox17 as gene_id: 20</span></li>
</ol><pre><code class="{bash} hljs">grep <span class="hljs-operator">-</span>o <span class="hljs-operator">-</span>i Sox17 exon_genes.gtf <span class="hljs-operator">|</span> wc <span class="hljs-operator">-</span>l
</code></pre><p><img src="https://i.imgur.com/2JZ64Gz.png" alt="" loading="lazy"></p><ol start="7">
<li><span>Number of “exon” in chr2: 29373</span></li>
</ol><pre><code class="{bash} hljs">awk <span class="hljs-string">'$1 = "chr2"'</span> <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>pipelines<span class="hljs-operator">/</span>references<span class="hljs-operator">/</span>genome_genes.gtf <span class="hljs-operator">&gt;</span> chr2_exon.gtf
</code></pre><pre><code class="{bash} hljs">wc <span class="hljs-operator">-</span>l chr2_exon.gtf
<span class="hljs-number">60606</span> chr2_exon.gtf
</code></pre><pre><code class="{bash} hljs">awk <span class="hljs-string">'$3 = "exon"'</span> chr2_exon.gtf <span class="hljs-operator">|</span> wc <span class="hljs-operator">-</span>l
</code></pre><p><img src="https://i.imgur.com/jUYWLWL.png" alt="" loading="lazy"></p><ol start="8">
<li><span>Frequency of each feature in column 3: 286783 CDS, 324748 exon, 29978 start_codon, 29953 stop_codon</span></li>
</ol><pre><code class="{bash} hljs">awk <span class="hljs-operator">-</span>F <span class="hljs-string">''</span><span class="hljs-string">'{print $3}'</span> <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>pipelines<span class="hljs-operator">/</span>references<span class="hljs-operator">/</span>genome_genes.gtf <span class="hljs-operator">|</span>sort <span class="hljs-operator">|</span> uniq <span class="hljs-operator">-</span>c
</code></pre><p><img src="https://i.imgur.com/3YQZFl3.png" alt="" loading="lazy"></p><ol start="9">
<li><span>Sort the file according to column1 and then column3. redirect the output into a new file</span></li>
</ol><pre><code class="{bash} hljs">sort <span class="hljs-operator">-</span>k <span class="hljs-number">1</span> <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>pipelines<span class="hljs-operator">/</span>references<span class="hljs-operator">/</span>genome_genes.gtf <span class="hljs-operator">&gt;</span> col1.gtf
</code></pre><pre><code class="{bash} hljs">sort <span class="hljs-operator">-</span>k <span class="hljs-number">3</span> col1.gtf <span class="hljs-operator">&gt;</span> col13.gtf
</code></pre><pre><code class="{bash} hljs">head col12.gtf
</code></pre><pre><code class="{bash} hljs">head col13.gtf
</code></pre><p><img src="https://i.imgur.com/q9SlliS.png" alt="" loading="lazy"></p><h2 id="5-Conda-and-Bioconda" data-id="5-Conda-and-Bioconda"><a class="anchor hidden-xs" href="#5-Conda-and-Bioconda" title="5-Conda-and-Bioconda"><span class="octicon octicon-link"></span></a><span>5 Conda and Bioconda</span></h2><p><span>a) Load the “base” environment if it is not loaded</span></p><pre><code class="{bash} hljs"><span class="hljs-keyword">source</span> <span class="hljs-regexp">/vol/</span>COMPEPIWS<span class="hljs-regexp">/conda/mi</span>niconda3<span class="hljs-regexp">/bin/</span>activate

</code></pre><p><span>b) create a new conda environment under your group conda directory called as your group’s name</span></p><pre><code class="{bash} hljs">conda create <span class="hljs-operator">-</span>p <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>groups<span class="hljs-operator">/</span>chipseq2<span class="hljs-operator">/</span>conda<span class="hljs-operator">/</span>chipseq2

</code></pre><p><span>c) activate the created environment</span></p><pre><code class="{bash} hljs">conda activate <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COPMPEPIWS<span class="hljs-operator">/</span>groups<span class="hljs-operator">/</span>chipseq2<span class="hljs-operator">/</span>conda<span class="hljs-operator">/</span>chipseq2

</code></pre><p><span>d) install ggplot2</span></p><pre><code class="{bash} hljs">conda install <span class="hljs-operator">-</span>c conda<span class="hljs-operator">-</span>forge r<span class="hljs-operator">-</span>ggplot2

</code></pre><p><span>install fastqc</span></p><pre><code class="{bash} hljs">conda <span class="hljs-keyword">install </span>-c <span class="hljs-keyword">bioconda </span>fastqc

</code></pre><p><span>install bedtools</span></p><pre><code class="{bash} hljs">conda <span class="hljs-keyword">install </span>-c <span class="hljs-keyword">bioconda </span><span class="hljs-keyword">bedtools=2.22
</span>
</code></pre><p><span>e) make sure the tools have been installed properly by :</span></p><pre><code class="{bash} hljs"><span class="hljs-meta">$ </span><span class="language-bash">fastqc --<span class="hljs-built_in">help</span></span>
</code></pre><pre><code class="{bash} hljs"><span class="hljs-meta">$ </span><span class="language-bash">bedtools --<span class="hljs-built_in">help</span></span>
</code></pre><pre><code class="{bash} hljs"><span class="hljs-variable">$</span> <span class="hljs-built_in">R</span>
</code></pre><p><span>f) update bedtools to the latest version, check the new version number</span></p><pre><code class="{bash} hljs">$ conda update -c <span class="hljs-keyword">bioconda </span><span class="hljs-keyword">bedtools
</span></code></pre><p><span>g) deactivate the environment</span></p><pre><code class="{bash} hljs"><span class="hljs-variable">$ </span>conda deactivate
</code></pre><h2 id="6-SLURM" data-id="6-SLURM"><a class="anchor hidden-xs" href="#6-SLURM" title="6-SLURM"><span class="octicon octicon-link"></span></a><span>6 SLURM</span></h2><ol>
<li><span>Create a new work folder and enter it</span></li>
</ol><pre><code class="{bash} hljs"><span class="hljs-built_in">mkdir</span> slurm
<span class="hljs-built_in">cd</span> slurm/

</code></pre><ol start="2">
<li><span>Run squeue and squeue -u </span>&lt;username&gt;<span>. Are there any jobs submitted by you? NO</span></li>
</ol><pre><code class="{bash} hljs">Run squeue <span class="hljs-keyword">and </span>squeue -u <span class="hljs-keyword">Shilpa
</span></code></pre><ol start="3">
<li><span>Submit jobs:</span></li>
</ol><pre><code class="{bash} hljs"><span class="hljs-keyword">for</span> i in $(seq <span class="hljs-number">1</span> <span class="hljs-number">15</span>)
do
sbatch <span class="hljs-operator">-</span>J test_${i} <span class="hljs-operator">-</span>o output_${i}.out <span class="hljs-operator">-</span><span class="hljs-operator">-</span>wrap<span class="hljs-operator">=</span><span class="hljs-string">"sleep 20 &amp;&amp; hostname"</span>
done
</code></pre><ol start="4">
<li><span>Check the job queue again</span></li>
</ol><pre><code class="{bash} hljs"><span class="hljs-attribute">squeue</span> -u Shilpa
</code></pre><ol start="5">
<li><span>When the job is done, compare the output to that of a local execution of hostname</span></li>
</ol><pre><code class="{bash} hljs"><span class="hljs-built_in">ls</span>
<span class="hljs-built_in">cat</span> output_14.out
<span class="hljs-variable">$</span> hostname
</code></pre><p><span>(Tells us that the tasks were executed on different nodes)</span><br>
<span>The node in the SLURM is bibigrid-worker-1-1-etqcgckjrdsvsxe</span><br>
<span>The node in local execution is bibigrid-master-etqcgckjrdsvsxe</span></p><h2 id="7-Basics-in-R" data-id="7-Basics-in-R"><a class="anchor hidden-xs" href="#7-Basics-in-R" title="7-Basics-in-R"><span class="octicon octicon-link"></span></a><span>7 Basics in R</span></h2><h4 id="Data-frames" data-id="Data-frames"><a class="anchor hidden-xs" href="#Data-frames" title="Data-frames"><span class="octicon octicon-link"></span></a><span>Data frames</span></h4><ol>
<li><span>activate the core conda environment and start R session</span></li>
</ol><pre><code class="{r} hljs">$ <span class="hljs-keyword">source</span> <span class="hljs-regexp">/vol/</span>COMPEPIWS<span class="hljs-regexp">/conda/mi</span>niconda3<span class="hljs-regexp">/bin/</span>activate <span class="hljs-regexp">/vol/</span>COMPEPIWS<span class="hljs-regexp">/conda/mi</span>niconda3<span class="hljs-regexp">/envs/</span>core

$ R
</code></pre><ol start="2">
<li><span>Load the following libraries: “ggplot2”, “reshape2” and “GenomicRanges”</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-function"><span class="hljs-title">library</span><span class="hljs-params">(ggplot2)</span></span>
<span class="hljs-function"><span class="hljs-title">library</span><span class="hljs-params">(reshape2)</span></span>
<span class="hljs-function"><span class="hljs-title">library</span><span class="hljs-params">(GenomicRanges)</span></span>
</code></pre><ol start="3">
<li><span>Read into R the following two tables: /vol/COMPEPIWS/groups/shared/*txt. Consider the first row as a header.</span></li>
</ol><p><span>#go to the shared folder:</span></p><pre><code class="{r} hljs"><span class="hljs-function"><span class="hljs-title">setwd</span><span class="hljs-params">(<span class="hljs-string">"/vol/COMPEPIWS/groups/shared/"</span>)</span></span>
</code></pre><p><span>#save the .txt files in a list:</span></p><pre><code class="{r} hljs">filelist <span class="hljs-operator">=</span> list.files(pattern <span class="hljs-operator">=</span> <span class="hljs-string">".txt"</span>)
</code></pre><p><span>#read:</span></p><pre><code class="{r} hljs">table1 <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> read.table(<span class="hljs-string">"/vol/COMPEPIWS/groups/shared/kidney_14.5_mouse_1_2kbW_bed_counts.txt"</span>, header <span class="hljs-operator">=</span> TRUE)

table2 <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> read.table(<span class="hljs-string">"/vol/COMPEPIWS/groups/shared/liver_14.5_mouse_1_2kbW_bed_counts.txt"</span>, header <span class="hljs-operator">=</span> TRUE)
</code></pre><ol start="4">
<li><span>The dimension of each table :</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-comment">#dim of table1:</span>
<span class="hljs-attribute">dim</span>(table1) : <span class="hljs-number">1362728</span> lines * <span class="hljs-number">6</span> columns 
<span class="hljs-comment">#dim of table2:</span>
<span class="hljs-attribute">dim</span>(table2) : <span class="hljs-number">1362728</span> lines * <span class="hljs-number">6</span> columns
</code></pre><ol start="5">
<li><span>Column names of each table :</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-selector-id">#col</span> names table1:
<span class="hljs-function"><span class="hljs-title">colnames</span><span class="hljs-params">(table1)</span></span> : <span class="hljs-string">"Chr"</span>      <span class="hljs-string">"Start"</span>    <span class="hljs-string">"End"</span>      <span class="hljs-string">"H3K27me3"</span> <span class="hljs-string">"H3K36me3"</span> <span class="hljs-string">"H3K9me3"</span>

<span class="hljs-selector-id">#col</span> names table2:
<span class="hljs-function"><span class="hljs-title">colnames</span><span class="hljs-params">(table2)</span></span> : <span class="hljs-string">"Chr"</span>      <span class="hljs-string">"Start"</span>    <span class="hljs-string">"End"</span>      <span class="hljs-string">"H3K27me3"</span> <span class="hljs-string">"H3K36me3"</span> <span class="hljs-string">"H3K9me3"</span>
</code></pre><ol start="6">
<li><span>Genome length from each data set : 2000</span></li>
</ol><pre><code class="{r} hljs">gen_len1 <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> table1$End <span class="hljs-operator">-</span> table1$Start
gen_len2 <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> table2$End <span class="hljs-operator">-</span> table2$Start
</code></pre><ol start="7">
<li><span>Create a data frame by concatenating vertically the two data sets. Add a new column called (cell_type) to annotate the rows of liver table with “liver” and the kidney’s rows with “kidney”.</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-comment">#add new col in table1:</span>
 table1[<span class="hljs-string">"cell_type"</span>] &lt;- <span class="hljs-string">"kidney"</span>
                        
</code></pre><pre><code class="{r} hljs">                        
<span class="hljs-comment">#add new col in table2:</span>
 table2[<span class="hljs-string">"cell_type"</span>] &lt;- <span class="hljs-string">"liver"</span>
</code></pre><pre><code class="{r} hljs"><span class="hljs-selector-id">#concatenate</span> table1 and table2 vertically:
<span class="hljs-selector-tag">table</span> &lt;- <span class="hljs-built_in">rbind</span>(table1, table2)
</code></pre><ol start="8">
<li><span>What is the dimension of the new data frame?</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-attribute">dim</span>(table): <span class="hljs-number">2725456</span> * <span class="hljs-number">7</span>
</code></pre><ol start="9">
<li><span>Reshape the data frame (excluding chr, start, end) from wide into long format using “cell_type” as variable id. What are the dimensions of this dataframe?:</span><br>
<span>Dimensions: 8176368 *  3</span></li>
</ol><pre><code class="{r} hljs">table3 <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> melt(table, id.vars <span class="hljs-operator">=</span> <span class="hljs-string">"cell_type"</span>, measure.vars <span class="hljs-operator">=</span> c(<span class="hljs-string">"H3K27me3"</span>, <span class="hljs-string">"H3K36me3"</span>, <span class="hljs-string">"H3K9me3"</span>))
</code></pre><pre><code class="{r} hljs"><span class="hljs-function"><span class="hljs-title">dim</span><span class="hljs-params">(table3)</span></span>
</code></pre><ol start="10">
<li><span>Plot the density of each variable considering the cell_type in two different panels. Give different transparent color for each variable (hint: ggplot, geom_density, facet_wrap). Save the plot as a pdf (hint: ggsave).</span></li>
</ol><pre><code class="{r} hljs">density_plot &lt;- ggplot(table3, aes(<span class="hljs-attribute">x</span>=enrichment, <span class="hljs-attribute">fill</span>=mark)) + 
         geom_density(<span class="hljs-attribute">alpha</span>=0.3) +
         facet_grid(cell_type~mark) +
         labs(<span class="hljs-attribute">x</span>=<span class="hljs-string">"Enrichment"</span>, <span class="hljs-attribute">y</span>=<span class="hljs-string">"Density"</span>)

ggsave(<span class="hljs-attribute">filename</span>=<span class="hljs-string">"density_plot.png"</span>, <span class="hljs-attribute">plot</span>=p_density, <span class="hljs-attribute">device</span>=<span class="hljs-string">"png"</span>)
</code></pre><p><img src="https://i.imgur.com/JbK0UxW.png" alt="" loading="lazy"></p><ol start="11">
<li><span>The plots look very skewed, try to set the limit of X axis to 100 and re-plot. Save the plot as a pdf.</span></li>
</ol><pre><code class="{r} hljs">scaled_density_plot &lt;- ggplot(table3, aes(<span class="hljs-attribute">x</span>=enrichment, <span class="hljs-attribute">fill</span>=mark)) + 
                geom_density(<span class="hljs-attribute">alpha</span>=0.3) +
                facet_grid(cell_type~mark) +
                labs(<span class="hljs-attribute">x</span>=<span class="hljs-string">"Enrichment"</span>, <span class="hljs-attribute">y</span>=<span class="hljs-string">"Density"</span>) +
                xlim(0, 100)

ggsave(<span class="hljs-attribute">filename</span>=<span class="hljs-string">"scaled_density_plot.png"</span>, <span class="hljs-attribute">plot</span>=p_scaled_density, <span class="hljs-attribute">device</span>=<span class="hljs-string">"pdf"</span>)
</code></pre><p><img src="https://i.imgur.com/PJPiVbh.png" alt="" loading="lazy"></p><ol start="12">
<li><span>Create a new data frame by concatenating horizontally the two data sets. Consider only the following columns: H3K27me3, H3K36me3 and H3K9me3.</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-selector-tag">table</span> &lt;- <span class="hljs-built_in">cbind</span>(<span class="hljs-built_in">select</span>(table1, <span class="hljs-built_in">c</span>(<span class="hljs-string">"H3K27me3"</span>, <span class="hljs-string">"H3K36me3"</span>, <span class="hljs-string">"H3K9me3"</span>)),
                   <span class="hljs-built_in">select</span>(table2,  <span class="hljs-built_in">c</span>(<span class="hljs-string">"H3K27me3"</span>, <span class="hljs-string">"H3K36me3"</span>, <span class="hljs-string">"H3K9me3"</span>)) )
</code></pre><ol start="13">
<li><span>Rename the header of the new data frame into: H3K27me3_liver, H3K36me3_liver, …</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-function"><span class="hljs-title">colnames</span><span class="hljs-params">(table)</span></span> &lt;- <span class="hljs-built_in">c</span>(<span class="hljs-string">"kidney_H3K27me3"</span>, <span class="hljs-string">"kidney_H3K36me3"</span>, <span class="hljs-string">"kidney_H3K9me3"</span>, <span class="hljs-string">"liver_H3K27me3"</span>, <span class="hljs-string">"liver_H3K36me3"</span>, <span class="hljs-string">"liver_H3K9me3"</span>)
</code></pre><ol start="14">
<li><span>Perform Principal Components Analysis on this data frame (hint: prcomp)</span></li>
</ol><pre><code class="{r} hljs">PCA <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> prcomp(table, scale<span class="hljs-operator">=</span>TRUE)
</code></pre><h4 id="Genomic-Ranges" data-id="Genomic-Ranges"><a class="anchor hidden-xs" href="#Genomic-Ranges" title="Genomic-Ranges"><span class="octicon octicon-link"></span></a><span>Genomic Ranges</span></h4><ol>
<li><span>Create two GenomicRange objects from the liver and kidney data frames using columns 4,5 and 6 as metadata</span></li>
</ol><pre><code class="{r} hljs">gr_kidney <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> GRanges(seqnames <span class="hljs-operator">=</span> table1$Chr,
                    ranges <span class="hljs-operator">=</span> IRanges(start<span class="hljs-operator">=</span>table1$Start, end<span class="hljs-operator">=</span>table1$End),
                    mcols<span class="hljs-operator">=</span> select(table1,c(<span class="hljs-string">"H3K27me3"</span>, <span class="hljs-string">"H3K36me3"</span>, <span class="hljs-string">"H3K9me3"</span>)))

gr_liver <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> GRanges(seqnames <span class="hljs-operator">=</span> table2$Chr,
                    ranges <span class="hljs-operator">=</span> IRanges(start<span class="hljs-operator">=</span>table2$Start, end<span class="hljs-operator">=</span>table2$End),
                    mcols<span class="hljs-operator">=</span> select(table2,c(<span class="hljs-string">"H3K27me3"</span>, <span class="hljs-string">"H3K36me3"</span>, <span class="hljs-string">"H3K9me3"</span>)))
</code></pre><ol start="2">
<li><span>Calculate the total number of bases covered in these two objects</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-function"><span class="hljs-title">coverage</span><span class="hljs-params">(gr_kidney)</span></span>
<span class="hljs-function"><span class="hljs-title">coverage</span><span class="hljs-params">(gr_liver)</span></span>
</code></pre><ol start="3">
<li><span>Subset one of the objects to have only chr2 lines</span></li>
</ol><pre><code class="{r} hljs">gr_kidney[seqnames(gr_kidney) <span class="hljs-operator">=</span><span class="hljs-operator">=</span> <span class="hljs-string">"chr2"</span>]
gr_liver[seqnames(gr_liver) <span class="hljs-operator">=</span><span class="hljs-operator">=</span> <span class="hljs-string">"chr2"</span>]
</code></pre><ol start="4">
<li><span>Shift one of the object’s ranges by 100 upstream</span></li>
</ol><pre><code class="{r} hljs">gr_kidney_shfit <span class="hljs-operator">&lt;</span><span class="hljs-operator">-</span> shift(gr_kidney, <span class="hljs-operator">+</span><span class="hljs-number">100</span>)
</code></pre><ol start="5">
<li><span>Find overlapping regions between the two objects</span></li>
</ol><pre><code class="{r} hljs">find<span class="hljs-constructor">Overlaps(<span class="hljs-params">gr_kidney_shfit</span>, <span class="hljs-params">gr_liver</span>)</span>
</code></pre><ol start="6">
<li><span>Make a GRangeList object from the two GRange objects</span></li>
</ol><pre><code class="{r} hljs"><span class="hljs-constructor">GRangesList(<span class="hljs-string">"Kidney"</span> = <span class="hljs-params">gr_kidney</span>, <span class="hljs-string">"Liver"</span> = <span class="hljs-params">gr_liver</span>)</span>
</code></pre><h1 id="ChIP-seq-Assay" data-id="ChIP-seq-Assay"><a class="anchor hidden-xs" href="#ChIP-seq-Assay" title="ChIP-seq-Assay"><span class="octicon octicon-link"></span></a><span>ChIP-seq Assay</span></h1><h2 id="Nextflow-ChIP-seq-pipeline" data-id="Nextflow-ChIP-seq-pipeline"><a class="anchor hidden-xs" href="#Nextflow-ChIP-seq-pipeline" title="Nextflow-ChIP-seq-pipeline"><span class="octicon octicon-link"></span></a><span>Nextflow ChIP-seq pipeline</span></h2><h3 id="1-Organize-your-raw-data" data-id="1-Organize-your-raw-data"><a class="anchor hidden-xs" href="#1-Organize-your-raw-data" title="1-Organize-your-raw-data"><span class="octicon octicon-link"></span></a><span>1. Organize your raw data</span></h3><p><span>a) Enter your working directory</span></p><pre><code class="{bash} hljs"><span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>groups<span class="hljs-operator">/</span><span class="hljs-operator">&lt;</span>group<span class="hljs-operator">&gt;</span><span class="hljs-operator">/</span>tasks
</code></pre><p><span>b) List the raw fastq files under</span></p><pre><code class="{bash} hljs"><span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>data<span class="hljs-operator">/</span>reduced<span class="hljs-operator">/</span>ChIP<span class="hljs-operator">-</span>seq
</code></pre><p><span>c) How many files per cell_type per time point per replicate are there?</span><br>
<span>32 files per cell type 16 files per time point  2 files per replicate</span></p><p><span>d) What are the histone marks you are dealing with?H3K27ac,H3K27me3,H3K4me1,H3K4me3,H3K9ac,H3K9me3,H3K36me3</span></p><p><span>e)  How many “control” per replicate do you have? : 2 or 4</span></p><p><span>f) Create a “data” folder in your group working directory and create symbolic links of the relevant files according to your group to the “data” folder as follow:</span></p><pre><code class="{bash} hljs">ln <span class="hljs-operator">-</span>s  <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>data<span class="hljs-operator">/</span>reduced<span class="hljs-operator">/</span>ChIP<span class="hljs-operator">-</span>seq<span class="hljs-comment">/*filename*  *symlinkname* 
</span></code></pre><h3 id="2-Generate-the-samplesheet" data-id="2-Generate-the-samplesheet"><a class="anchor hidden-xs" href="#2-Generate-the-samplesheet" title="2-Generate-the-samplesheet"><span class="octicon octicon-link"></span></a><span>2. Generate the samplesheet</span></h3><p><span>a) Enter the task folder</span></p><pre><code class="{bash} hljs"><span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>groups<span class="hljs-operator">/</span>chipseq2<span class="hljs-operator">/</span>tasks
</code></pre><p><span>b) Prepare the samplesheet</span></p><pre><code class="{bash} hljs"><span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> /vol/COMPEPIWS/data/reduced/ChIP-seq/*liver*gz
<span class="hljs-keyword">do</span>
group=$(<span class="hljs-built_in">basename</span> <span class="hljs-variable">${i}</span>|<span class="hljs-built_in">cut</span> -f 1-3 -d _)
replicate=$(<span class="hljs-built_in">basename</span> <span class="hljs-variable">${i}</span>|<span class="hljs-built_in">cut</span> -f 4 -d _)
fastq1=<span class="hljs-variable">$i</span>
fastq2=<span class="hljs-string">''</span>
<span class="hljs-keyword">if</span> [ <span class="hljs-string">"<span class="hljs-subst">$(basename ${i}|cut -f 3 -d _)</span>"</span> == <span class="hljs-string">"control"</span> ];
<span class="hljs-keyword">then</span>
mark=<span class="hljs-string">''</span>
control=<span class="hljs-string">''</span>
<span class="hljs-keyword">else</span>
mark=$(<span class="hljs-built_in">basename</span> <span class="hljs-variable">${i}</span>|<span class="hljs-built_in">cut</span> -f 3 -d _)
control=$(<span class="hljs-built_in">basename</span> <span class="hljs-variable">${i}</span>|<span class="hljs-built_in">cut</span> -f 1-2 -d _)<span class="hljs-string">"_control"</span> 
<span class="hljs-keyword">fi</span>
<span class="hljs-built_in">echo</span> <span class="hljs-string">"<span class="hljs-variable">$group</span>,<span class="hljs-variable">$replicate</span>,<span class="hljs-variable">$fastq1</span>,,<span class="hljs-variable">$mark</span>,<span class="hljs-variable">$control</span>"</span>
<span class="hljs-keyword">done</span> &gt; samplesheet.csv
</code></pre><h3 id="3-Running-the-pipeline" data-id="3-Running-the-pipeline"><a class="anchor hidden-xs" href="#3-Running-the-pipeline" title="3-Running-the-pipeline"><span class="octicon octicon-link"></span></a><span>3. Running the pipeline</span></h3><p><span>a) Based on the config file:</span></p><pre><code>/vol/COMPEPIWS/pipelines/configs/chipseq.config.
</code></pre><p><span>i) Single end</span><br>
<span>ii) differential analysis,preseq,plot_profile,plot_fingerprint,spp</span><br>
<span>iii) Blacklist regions account for a significant portion of ChIP-seq reads, are driven by artifacts in genome assemblies, and removal of these regions is essential to removing noise in genomics assays.</span></p><p><span>Regions present : chr 18 &amp; 19</span></p><p><span>iv) Tool used for mapping : BWA</span></p><p><span>b) Load Conda environment:</span></p><pre><code class="{bash} hljs"><span class="hljs-keyword">source</span><span class="hljs-regexp">/vol/</span>COMPEPIWS<span class="hljs-regexp">/conda/mi</span>niconda3<span class="hljs-regexp">/bin/</span>activate<span class="hljs-regexp">/vol/</span>COMPEPIWS<span class="hljs-regexp">/conda/mi</span>niconda3<span class="hljs-regexp">/envs/</span>core
</code></pre><p><span>c) Run the pipeline</span></p><pre><code class="{bash} hljs">nextflow run nf<span class="hljs-operator">-</span>core<span class="hljs-operator">/</span>chipseq <span class="hljs-operator">-</span>r <span class="hljs-number">1.2</span><span class="hljs-number">.2</span> <span class="hljs-operator">-</span>profile singularity <span class="hljs-operator">-</span>c <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>pipelines<span class="hljs-operator">/</span>configs<span class="hljs-operator">/</span>chipseq.config <span class="hljs-operator">-</span><span class="hljs-operator">-</span>input samplesheet.csv
</code></pre><ol start="4">
<li><span>To watch our jobs</span></li>
</ol><pre><code>squeue --user=$ Shilpa
</code></pre><ol start="5">
<li><span>The output folder with the name “results”</span></li>
</ol><pre><code>/vol/COMPEPIWS/groups/chipseq2/results
</code></pre><ol start="6">
<li><span>Main steps of the pipeline for processing ChIP-seq :</span><br>
<span>Raw data,fastqc,Trimming,mapping (bwa),peak calling(MACS2)</span></li>
</ol><h2 id="2-Quality-Control" data-id="2-Quality-Control"><a class="anchor hidden-xs" href="#2-Quality-Control" title="2-Quality-Control"><span class="octicon octicon-link"></span></a><span>2 Quality Control</span></h2><p><span>2.1  Exploring the result folder</span><br>
<span>1.</span></p><pre><code>/vol/COMPEPIWS/groups/chipseq2/results/pipeline_info/execution_report.html
</code></pre><p><span>a) Time taken to pipeline to finish : 2h 5m 35s</span><br>
<span>b) Step which took the most CPU/Memory usage : BWA_MEM (Mapping)</span><br>
<span>2. Aligned reads are located in :</span></p><pre><code>/vol/COMPEPIWS/groups/chipseq2/results/bwa/mergedLibrary
</code></pre><ol start="3">
<li><span>Peak files are loacted in :</span></li>
</ol><pre><code>/vol/COMPEPIWS/groups/chipseq2/results/bwa/mergedLibrary/macs/narrowPeak
</code></pre><p><span>Tool used for Peak Calling: MACS2</span><br>
<span>4. Consensus peaks are located in :</span></p><pre><code>/vol/COMPEPIWS/groups/chipseq2/results/bwa/mergedLibrary/macs/narrowPeak/consensus
</code></pre><ol start="5">
<li><span>Signal tracks are located in:</span></li>
</ol><pre><code>/vol/COMPEPIWS/groups/chipseq2/results/bwa/mergedLibrary/bigwig
</code></pre><p><span>2.2 MultiQC</span></p><ol>
<li><span>To explore diferent section of the MultiQC HTML report generated by the pipeline</span></li>
</ol><pre><code>vol/COMPEPIWS/groups/chipseq2/results/multiqc/narrowPeak
</code></pre><p><img src="https://i.imgur.com/q4mWQKn.png" alt="" loading="lazy"></p><ol start="2">
<li></li>
</ol><p><span>The quality score represents the probability that the corresponding nucleotide or base in the read is called incorrectly represented. It is calculated as Q = -10 x log10 P, where P is the probability that a base call is erroneous.From the figure below, note that there is a base 12 that has score around 23 which is lying below the green region. This indicates that base 12 of sample liver_14.5_H3K4me1_R2_T1 has poor quality, while rest of the bases of most of the samples are lying above green region and hence are of good quality.</span></p><p><img src="https://i.imgur.com/63jaZdH.png" alt="" loading="lazy"></p><p><span>We can see from the graph below that there are more number of reads for most of the samples that has average quality score more than 30. However, there are also fraction of reads of samples which are having average quality score is less than 30 (or lying in the red zone).</span></p><p><img src="https://i.imgur.com/REbLT5w.png" alt="" loading="lazy"></p><p><span>The figure below represents the graph of “number of reads” vs “sequence length”. For instance we can see that liver_15.5_control_R2_T1 sample has more than 3M reads with sequence length of 50bp.</span><br>
<img src="https://i.imgur.com/5MpYUM7.png" alt="" loading="lazy"></p><p><span>Adapter contamination:</span><br>
<img src="https://i.imgur.com/zFRgthR.png" alt="" loading="lazy"></p><p><span>From given two figures below, we can note that the variable “number of reads” is changed. Note that, for the mentioned sample in figures, number of unique reads is increased while number of duplicate reads is decreased in “after trimming” as compared to “before trimming”. This might be due to the fact that some of the duplicate reads after trimming become unique and hence number of unique reads is increased, though increase is not significant for the mentioned sample. Furthermore, we have lost some reads after trimming of around 100 of reads for the given sample in figures.</span><br>
<span>number of reads “before trimming (raw)”:</span><br>
<img src="https://i.imgur.com/6WuLcaM.png" alt="" loading="lazy"></p><p><span>number of reads “after trimming”:</span><br>
<img src="https://i.imgur.com/WdXFte7.png" alt="" loading="lazy"></p><ol start="3">
<li><span>Mapping efficiency: Mapping efficiency decreases with the increase of # unmapped reads as it give rise to low confidence on the score we calculate for further analysis. Note from the figure below, the sample liver_14.5_H3K4me3_R2 has high percentage of unmapped reads as compare to other samples (in terms of unmapped reads).Therefore this sample has comparatively low mapping efficiency.</span><br>
<img src="https://i.imgur.com/7P6O3dI.png" alt="" loading="lazy"></li>
</ol><p><span>Duplication rate : If we have more duplicate reads then that will also affect our mapping quality. Large number of duplicates =&gt; high duplication rate. For instance, in the graph below, sample liver_14.5_H3K27ac_R2 has highest number of duplicate reads and therefore high duplication rate.</span><br>
<span>Q) what does these staright lines indicate in “Paired in sequencing” etc.?</span><br>
<img src="https://i.imgur.com/2e3Ubac.png" alt="" loading="lazy"></p><ol start="4">
<li>
<p><span>Do you get the similar number of peaks for all histone marks? : No. From the figure below, we can clearly compare that there are some marks with significantly larger number of peaks than other histone marks.</span><br>
<img src="https://i.imgur.com/xZtNzdA.png" alt="" loading="lazy"></p>
<p><span>Overall:</span><br>
<span>Marks with the highest number of peaks : H3K4me1</span><br>
<span>Marks with the lowest number of peaks : H3K27me3</span></p>
</li>
</ol><p><span>5.What does FRiP score mean?</span><br>
<span>It is generated by calculating the fraction of all mapped   reads that fall into the MACS2 called peak regions. A  read must overlap a peak by at least 20% to be counted.</span></p><p><span>Which marks get the highest/lowest FRiP scores?</span><br>
<img src="https://i.imgur.com/LlJlBCZ.png" alt="" loading="lazy"><br>
<span>Marks with highest FRiP scores : H3K4me3 (0.43)</span><br>
<span>Marks with lowest FRiP scores : H3K4me1 (0.02)</span></p><p><span>How is it related to the mark signal distribution (narrow/broad)?</span><br>
<span>higher the frip score =&gt; broader the signal</span></p><p><span>2.3 deepTools2</span></p><ol>
<li><span>Activate the core environment</span></li>
</ol><pre><code>source /vol/COMPEPIWS/conda/miniconda3/bin/activate /vol/COMPEPIWS/conda/miniconda3/envs/core
</code></pre><ol start="2">
<li><span>To Explore the batch effect between the samples</span></li>
</ol><pre><code>multiBigwigSummary bins -b liver_14.5_H3K27ac_R1.bigWig liver_14.5_H3K27ac_R2.bigWig liver_14.5_H3K27me3_R1.bigWig liver_14.5_H3K27me3_R2.bigWig liver_14.5_H3K36me3_R1.bigWig liver_14.5_H3K36me3_R2.bigWig liver_14.5_H3K4me1_R1.bigWig liver_14.5_H3K4me1_R2.bigWig liver_14.5_H3K4me3_R1.bigWig liver_14.5_H3K4me3_R2.bigWig liver_14.5_H3K9ac_R1.bigWig liver_14.5_H3K9ac_R2.bigWig liver_14.5_H3K9me3_R1.bigWig liver_14.5_H3K9me3_R2.bigWig liver_14.5_control_R1.bigWig liver_14.5_control_R2.bigWig liver_15.5_H3K27ac_R1.bigWig liver_15.5_H3K27ac_R2.bigWig liver_15.5_H3K27me3_R1.bigWig liver_15.5_H3K27me3_R2.bigWig liver_15.5_H3K36me3_R1.bigWig liver_15.5_H3K36me3_R2.bigWig liver_15.5_H3K4me1_R1.bigWig liver_15.5_H3K4me1_R2.bigWig liver_15.5_H3K4me3_R1.bigWig liver_15.5_H3K4me3_R2.bigWig liver_15.5_H3K9ac_R1.bigWig liver_15.5_H3K9ac_R2.bigWig liver_15.5_H3K9me3_R1.bigWig liver_15.5_H3K9me3_R2.bigWig liver_15.5_control_R1.bigWig liver_15.5_control_R2.bigWig -o chipseq2results.npz
</code></pre><p><span>number of bins found :15215</span></p><pre><code class="{bash} hljs"><span class="hljs-attribute">plotCorrelation</span> -h
</code></pre><pre><code class="{bash} hljs">plotCorrelation <span class="hljs-operator">-</span>in chipseq2results.npz <span class="hljs-operator">-</span><span class="hljs-operator">-</span>whatToPlot heatmap <span class="hljs-operator">-</span><span class="hljs-operator">-</span>corMethod pearson <span class="hljs-operator">-</span>o heatmap.png
</code></pre><p><img src="https://i.imgur.com/5SjRp3D.png" alt="" loading="lazy"></p><pre><code class="{bash} hljs">plotPCA <span class="hljs-operator">-</span>in chipseq2results.npz <span class="hljs-operator">-</span>o pca.png
</code></pre><p><img src="https://i.imgur.com/aa3ZnLV.png" alt="" loading="lazy"></p><ol start="3">
<li>
<p><span>Marks that are similar to each other based on plotCorrelation outputs</span><br>
<span>H3K9ac ~ H3K4me3</span><br>
<span>H3K9ac ~ H3K4me1</span><br>
<span>H3K36me3 !~ H3K27me3</span></p>
</li>
<li>
<p><span>To separate signals (marks) from background:</span><br>
<span>PATH &gt; /vol/COMPEPIWS/groups/chipseq2/results/bwa/mergedLibrary/ to fingerprint.png</span></p>
</li>
</ol><pre><code class="{bash} hljs">fingerprint: plotFingerprint -<span class="hljs-selector-tag">b</span> liver_14.<span class="hljs-number">5</span>_H3K9me3_R1<span class="hljs-selector-class">.mLb</span><span class="hljs-selector-class">.clN</span><span class="hljs-selector-class">.sorted</span><span class="hljs-selector-class">.bam</span> liver_14.<span class="hljs-number">5</span>_H3K9me3_R2<span class="hljs-selector-class">.mLb</span><span class="hljs-selector-class">.clN</span><span class="hljs-selector-class">.sorted</span><span class="hljs-selector-class">.bam</span> liver_15.<span class="hljs-number">5</span>_H3K9me3_R1<span class="hljs-selector-class">.mLb</span><span class="hljs-selector-class">.clN</span><span class="hljs-selector-class">.sorted</span><span class="hljs-selector-class">.bam</span> liver_15.<span class="hljs-number">5</span>_H3K9me3_R2<span class="hljs-selector-class">.mLb</span><span class="hljs-selector-class">.clN</span><span class="hljs-selector-class">.sorted</span><span class="hljs-selector-class">.bam</span> -plot fingerprint_H3K9me3<span class="hljs-selector-class">.png</span>

plotFingerprint -<span class="hljs-selector-tag">b</span> liver_14.<span class="hljs-number">5</span>_H3K9me3_R1<span class="hljs-selector-class">.mLb</span><span class="hljs-selector-class">.clN</span><span class="hljs-selector-class">.sorted</span><span class="hljs-selector-class">.bam</span> liver_14.<span class="hljs-number">5</span>_H3K9me3_R2<span class="hljs-selector-class">.mLb</span><span class="hljs-selector-class">.clN</span><span class="hljs-selector-class">.sorted</span><span class="hljs-selector-class">.bam</span> liver_15.<span class="hljs-number">5</span>_H3K9me3_R1<span class="hljs-selector-class">.mLb</span><span class="hljs-selector-class">.clN</span><span class="hljs-selector-class">.sorted</span><span class="hljs-selector-class">.bam</span> liver_15.<span class="hljs-number">5</span>_H3K9me3_R2<span class="hljs-selector-class">.mLb</span><span class="hljs-selector-class">.clN</span><span class="hljs-selector-class">.sorted</span><span class="hljs-selector-class">.bam</span> -plot fingerprint_H3K9me3<span class="hljs-selector-class">.png</span>
</code></pre><p><img src="https://i.imgur.com/htBLBct.png" alt="" loading="lazy"></p><p><img src="https://i.imgur.com/QHNg3n9.png" alt="" loading="lazy"><br>
<span>Not well separated from background for histone mark H3K9me3.</span></p><ol start="5">
<li><span>Marks enriched at TSS : H3K4me3, H3K9ac</span></li>
</ol><pre><code class="{bash} hljs">computeMatrix scale<span class="hljs-operator">-</span>regions <span class="hljs-operator">-</span>S liver_14.5_H3K27ac_R1.bigWig liver_14.5_H3K27me3_R1.bigWig liver_14.5_H3K36me3_R1.bigWig liver_14.5_H3K4me1_R1.bigWig liver_14.5_H3K4me3_R1.bigWig liver_14.5_H3K9ac_R1.bigWig liver_14.5_H3K9me3_R1.bigWig <span class="hljs-operator">-</span>R <span class="hljs-operator">/</span>vol<span class="hljs-operator">/</span>COMPEPIWS<span class="hljs-operator">/</span>groups<span class="hljs-operator">/</span>chipseq2<span class="hljs-operator">/</span>results<span class="hljs-operator">/</span>genome<span class="hljs-operator">/</span>genome_genes.bed <span class="hljs-operator">-</span>a <span class="hljs-number">1500</span> <span class="hljs-operator">-</span>b <span class="hljs-number">1500</span> <span class="hljs-operator">-</span>o chipseq2matrix

plotProfile <span class="hljs-operator">-</span>m chipseq2matrix <span class="hljs-operator">-</span>o profileplot.png
</code></pre><p><img src="https://i.imgur.com/xJAHCc7.png" alt="" loading="lazy"></p><p><span>PATH &gt; /vol/COMPEPIWS/groups/chipseq2/results/bwa/mergedLibrary/bigwig/ to profileplot.png and heatmap.png</span></p><h2 id="3-Exploratory-Analysis-using-IGV" data-id="3-Exploratory-Analysis-using-IGV"><a class="anchor hidden-xs" href="#3-Exploratory-Analysis-using-IGV" title="3-Exploratory-Analysis-using-IGV"><span class="octicon octicon-link"></span></a><span>3  Exploratory Analysis using IGV</span></h2><ol>
<li><span>IGV installation</span></li>
</ol><pre><code>/vol/COMPEPIWS/softwares/IGV_Linux_2.10.2/
</code></pre><ol start="2">
<li><span>To start IGV</span></li>
</ol><pre><code>/vol/COMPEPIWS/softwares/IGV_Linux_2.10.2/igv.sh
</code></pre><ol start="3">
<li><span>To open IGV session from :</span></li>
</ol><pre><code>results/igv/narrowPeak/igv_session.xml
</code></pre><p><span>To remove consensus tracks : Right-click a track name and select Remove Tracks in the pop-up menu</span></p><ol start="4">
<li><span>To upload gene tracks</span></li>
</ol><pre><code>results/genome/genome_genes.bed
</code></pre><p><img src="https://i.imgur.com/XUMRpND.png" alt="" loading="lazy"></p><ol start="5">
<li>
<p><span>Do you see clear peaks for all marks? : Yes</span></p>
<p><span>Sharper marks : H3K9ac,H3K4me3,H3K4me1,H3K36me3,H3K27me3,H3K27ac</span></p>
<p><img src="https://i.imgur.com/yfYH3q9.png" alt="" loading="lazy"></p>
<p><span>Broader marks : H3K9me3 &amp; Control</span></p>
<p><img src="https://i.imgur.com/y4qz4yI.png" alt="" loading="lazy"></p>
</li>
<li>
<p><span>Coloring the tracks as we want (e.g. per mark). Right click on the track name.</span></p>
<p><img src="https://i.imgur.com/QvETQSV.png" alt="" loading="lazy"></p>
</li>
<li>
<p><span>Marks that overlap with TSS : H3K27ac, H3K4me3, H3K9ac</span><br>
<span>Marks that overlap with gene body : H3K9ac, H3K4me1, H3K36me3, H3K27me3</span></p>
<p><img src="https://i.imgur.com/CakzSAJ.png" alt="" loading="lazy"></p>
</li>
<li>
<p><span>Do you always see an agreement between the called peaks of the two replicates of the same mark in the same cell type? What do you understand from this plot (choose two of the sharp marks)</span></p>
</li>
</ol><p><span>No, this is always not the case. As we can see in the figure below that for the mark H3K27ac R1 and R2 are not agreeing on the same region. This may arise due to the signal noise or the may be due to contamination of the solution while preparing the sample.</span></p><p><img src="https://i.imgur.com/LUfoX9K.png" alt="" loading="lazy"><br>
<img src="https://i.imgur.com/xwqTpAf.png" alt="" loading="lazy"></p><h2 id="4-Chromatin-segmentation-with-ChromHMM" data-id="4-Chromatin-segmentation-with-ChromHMM"><a class="anchor hidden-xs" href="#4-Chromatin-segmentation-with-ChromHMM" title="4-Chromatin-segmentation-with-ChromHMM"><span class="octicon octicon-link"></span></a><span>4 Chromatin segmentation with ChromHMM</span></h2><p><span>4.1 ChromHMM Installation</span></p><pre><code>/vol/COMPEPIWS/softwares/ChromHMM/
</code></pre><p><span>4.2 Running ChromHMM</span></p><ol>
<li><span>Activate Core conda environment</span></li>
</ol><pre><code>source /vol/COMPEPIWS/conda/miniconda3/bin/activate /vol/COMPEPIWS/conda/miniconda3/envs/core
</code></pre><ol start="2">
<li><span>Create a folder called segmentation under your work group folder, and enter it</span></li>
</ol><pre><code>mkdir segmentation
</code></pre><ol start="3">
<li><span>Create a folder called “inputs”</span></li>
</ol><pre><code>mkdir inputs
</code></pre><p><span>Create symbolic links of all aligned read files to this folder (there should be 64 bam files).</span></p><pre><code>for i in /vol/COMPEPIWS/groups/chipseq2/results/bwa/mergedLibrary/*liver*bam; do ln -s $i /vol/COMPEPIWS/groups/chipseq2/segmentation/inputs/ ; done
</code></pre><ol start="4">
<li><span>Prepare a tab delimited file “cellmarkfiletable.txt”</span></li>
</ol><pre><code>for i in *liver*bam *kidney*bam 
do
cell_type=$(basename ${i}|cut -f 1-2 -d _)
histone_mark=$(basename ${i}|cut -f 3 -d _)
celltype_stage=$(basename ${i}|cut -f 1-4 -d _)
if [ "$(basename ${i}|cut -f 1-2 -d _)" == "liver_14.5" ];
then
control='liver_14.5_control_R1.mLb.clN.sorted.bam'
elif [ "$(basename ${i}|cut -f 1-2 -d _)" == "liver_15.5" ];
then
control='liver_15.5_control_R1.mLb.clN.sorted.bam'
elif [ "$(basename ${i}|cut -f 1-2 -d _)" == "kidney_14.5" ];
then
control='kidney_14.5_control_R1.mLb.clN.sorted.bam'
elif [ "$(basename ${i}|cut -f 1-2 -d _)" == "kidney_15.5" ];
then
control='kidney_15.5_control_R1.mLb.clN.sorted.bam'
fi
printf '%s\t%s\t%s\t%s\n' "$cell_type" "$histone_mark" "$celltype_stage" "$control"
done &gt; cellmarkfiletable.txt
</code></pre><ol start="5">
<li><span>Run the BinarizeBam command with 200bp as a bin size</span></li>
</ol><pre><code>java -Xmx3000M -Djava.awt.headless=true -jar /vol/COMPEPIWS/softwares/ChromHMM/ChromHMM.jar BinarizeBam -b 200 /vol/COMPEPIWS/softwares/ChromHMM/CHROMSIZES/mm10.txt /vol/COMPEPIWS/groups/chipseq2/segmentation/inputs inputs/cellmarkfiletable.txt outputs
</code></pre><ol start="6">
<li><span>Run the LearnModel command with different number of states: 5-15</span></li>
</ol><pre><code>for i in $(seq 5 15)
do
sbatch --job-name=learnmodel -o output_${i}.out --wrap="java -Xmx3000M -Djava.awt.headless=true -jar /vol/COMPEPIWS/softwares/ChromHMM/ChromHMM.jar LearnModel /vol/COMPEPIWS/groups/chipseq2/segmentation/outputs /vol/COMPEPIWS/groups/chipseq2/segmentation/learnmodel_out $i mm10"
done
</code></pre><ol start="7">
<li><span>Explore the output files called webpage_$state.html. Especially, emission heatmap, Ref Seq_TSS_neighborhood and overlap figures. Comment on the “new” states you get.</span></li>
</ol><p><img src="https://i.imgur.com/CEFKOOH.png" alt="" loading="lazy"></p><p><img src="https://i.imgur.com/cfCQELY.png" alt="" loading="lazy"></p><p><img src="https://i.imgur.com/4sH7lyN.png" alt="" loading="lazy"></p><ol start="8">
<li><span>Compare the different models to assess what is the proper minimum number of states: 10 number of states are appropriate for</span></li>
</ol><pre><code>java -Xmx3000M -Djava.awt.headless=true -jar /vol/COMPEPIWS/softwares/ChromHMM/ChromHMM.jar CompareModels emissions_15.txt /vol/COMPEPIWS/groups/chipseq2/segmentation/learnmodel_out/ compare_model
</code></pre><ol start="9">
<li><span>Repeat step 5 to merge the stages of each cell type in order to produce only 2 segmentation files (1 for each cell type).</span></li>
</ol><pre><code>java -Xmx3000M -Djava.awt.headless=true -jar /vol/COMPEPIWS/softwares/ChromHMM/ChromHMM.jar BinarizeBam -b 200 /vol/COMPEPIWS/softwares/ChromHMM/CHROMSIZES/mm10.txt /vol/COMPEPIWS/groups/chipseq2/segmentation/inputs segmentation/cellmarkfiletable.txt merge_outputs
</code></pre><ol start="10">
<li><span>Repeat steps 6 with the new “cellmarkfiletable” for only 15 states model</span></li>
</ol><pre><code>sbatch --job-name=learnmodel -o output_${i}.out --wrap="java -Xmx3000M -Djava.awt.headless=true -jar /vol/COMPEPIWS/softwares/ChromHMM/ChromHMM.jar LearnModel /vol/COMPEPIWS/groups/chipseq2/merge_outputs /vol/COMPEPIWS/groups/chipseq2/segmentation/learnmodel_merge_out 15 mm10"
</code></pre><ol start="11">
<li>
<p><span>Based on the emission heatmap, overlap and enrichment of this new 15-state model, assigning biological</span><br>
<span>labels to the states</span><br>
<img src="https://i.imgur.com/KMdp99B.png" alt="" loading="lazy"><br>
<img src="https://i.imgur.com/9ZaY8Fg.png" alt="" loading="lazy"></p>
</li>
<li>
<p><span>Rename the states in column 4 of the files *dense.bed (or *segments.bed) using the biological states from the previous question (consider the command MakeBrowserFiles from ChromHMM or awk in bash to do the task). Note: Redirect the output into new files in a subfolder called “relabeled” to avoid overwriting the original files.</span><br>
<span>For liver:</span></p>
</li>
</ol><pre><code>java -Xmx3000M -Djava.awt.headless=true -jar /vol/COMPEPIWS/softwares/ChromHMM/ChromHMM.jar MakeBrowserFiles -m /vol/COMPEPIWS/groups/chipseq2/segmentation/seg_name.txt /vol/COMPEPIWS/groups/chipseq2/segmentation/learnmodel_merge_out/liver_15_segments.bed seg_chipseq2 /vol/COMPEPIWS/groups/chipseq2/segmentation/relabelled/liver_relabelled
</code></pre><p><span>For kidney :</span></p><pre><code>java -Xmx3000M -Djava.awt.headless=true -jar /vol/COMPEPIWS/softwares/ChromHMM/ChromHMM.jar MakeBrowserFiles -m /vol/COMPEPIWS/groups/chipseq2/segmentation/seg_name.txt /vol/COMPEPIWS/groups/chipseq2/segmentation/learnmodel_merge_out/kidney_15_segments.bed seg_chipseq2 /vol/COMPEPIWS/groups/chipseq2/segmentation/relabelled/kidney_relabelled
</code></pre><ol start="13">
<li><span>Generating the emission heatmap with the new labels using the command “reorder”</span></li>
</ol><pre><code>java -Xmx3000M -Djava.awt.headless=true -jar /vol/COMPEPIWS/softwares/ChromHMM/ChromHMM.jar Reorder -m /vol/COMPEPIWS/groups/chipseq2/segmentation/seg_name.txt /vol/COMPEPIWS/groups/chipseq2/segmentation/learnmodel_merge_out/model_15.txt /vol/COMPEPIWS/groups/chipseq2/segmentation/relabelled/output  
</code></pre><ol start="14">
<li><span>Copy the relabeled segmentation files of Q12 and the emission heatmap of Q13 into the shared folder under</span><br>
<span>your group sub-directory /vol/COMPEPIWS/groups/shared/</span>&lt;group&gt;<span>/segmentation</span></li>
</ol><pre><code>for i in /vol/COMPEPIWS/groups/chipseq2/segmentation/relabelled/*.bed; do ln -s $i /vol/COMPEPIWS/groups/shared/ChIP-seq/chipseq2/segmentation; done
    
for i in /vol/COMPEPIWS/groups/chipseq2/segmentation/relabelled/output/*emissions*png; do ln -s $i /vol/COMPEPIWS/groups/shared/ChIP-seq/chipseq2/segmentation; done
</code></pre><h2 id="5-Exploration-of-chromatin-state-segmentation-using-IGV" data-id="5-Exploration-of-chromatin-state-segmentation-using-IGV"><a class="anchor hidden-xs" href="#5-Exploration-of-chromatin-state-segmentation-using-IGV" title="5-Exploration-of-chromatin-state-segmentation-using-IGV"><span class="octicon octicon-link"></span></a><span>5 Exploration of chromatin state segmentation using IGV</span></h2><ol>
<li>
<p><span>Which state(s) is(are) more likely overlapping with TSS?</span><br>
<img src="https://i.imgur.com/2cUWWpZ.png" alt="" loading="lazy"></p>
</li>
<li>
<p><span>Some differential staes in kidney and liver samples.</span><br>
<img src="https://i.imgur.com/4tni12B.png" alt="" loading="lazy"></p>
</li>
</ol><h2 id="6-Exploratory-Analysis-using-R" data-id="6-Exploratory-Analysis-using-R"><a class="anchor hidden-xs" href="#6-Exploratory-Analysis-using-R" title="6-Exploratory-Analysis-using-R"><span class="octicon octicon-link"></span></a><span>6 Exploratory Analysis using R</span></h2><ol>
<li><span>Activate the “core” environment.</span></li>
</ol><pre><code>source /vol/COMPEPIWS/conda/miniconda3/bin/activate /vol/COMPEPIWS/conda/miniconda3/envs/core
Start R
</code></pre><ol start="2">
<li><span>Start R</span></li>
<li><span>Read all *(reordered)dense.bed files from task 4.2.12 from “segmentation” directory results into R. (hint:</span><br>
<span>use list.files, lapply, read.table functions)</span></li>
</ol><pre><code>dir &lt;-"vol/COMPEPIWS/groups/chipseq2/segmentation/relabelled/"
regex &lt;- "*dense.bed$"
filenames &lt;- list.files(path=dir,
                        pattern=regex,
                        full.names=TRUE)
filenames
BEDs &lt;- lapply(filenames,
               read.table,
               header=FALSE,
               sep="\t",
               stringsAsFactors=FALSE,
               skip=1)

BEDs[[1]]$sample &lt;- "kidney"
BEDs[[2]]$sample &lt;- "liver"
</code></pre><ol start="4">
<li><span>Merge the results into one data frame (hint: rbind function)</span></li>
</ol><pre><code>df &lt;- rbind(BEDs[[1]], BEDs[[2]])
</code></pre><h1 id="Integrative-analysis" data-id="Integrative-analysis"><a class="anchor hidden-xs" href="#Integrative-analysis" title="Integrative-analysis"><span class="octicon octicon-link"></span></a><span>Integrative analysis</span></h1><h2 id="2-Integrative-data-exploration-in-IGV" data-id="2-Integrative-data-exploration-in-IGV"><a class="anchor hidden-xs" href="#2-Integrative-data-exploration-in-IGV" title="2-Integrative-data-exploration-in-IGV"><span class="octicon octicon-link"></span></a><span>2. Integrative data exploration in IGV</span></h2><ol>
<li><span>What are the methylation states of the promoters and the gene body in general (high, low)?</span><br>
<span>In the figure below (zoomed in), we can see that the region of interest (genomic region: chr18:31,755,635-31,775,374) has an unmethylated promoter  for both kidney and liver sequences. While the gene body corresponding to this promoter region is highly methylated (HMR)  to maintain the transcription process and hinder the binding of other proteins to the gene body. Moreover, gene body represents strong transcription (Tx_S).</span></li>
</ol><p><img src="https://i.imgur.com/tDAjegn.png" alt="" loading="lazy"></p><p><span>For an other instance (figure given below), it is clearly visible for the full length of gene ‘Slc25a46’ that promoters regions for kidney and liver samples are mostly unmethylated while their gene body lie under HMR.</span><br>
<img src="https://i.imgur.com/9Tohbh6.png" alt="" loading="lazy"></p><p><span>So in general, we can conclude that the promoter regions lie under UMR while their corresponding gene body lies under HMR to maintain the transcription process and complete it successfully.</span><br>
<span>Note: Regions representing Tx_S as well as Tx_W are both HMR so we observe that level of trancription does not affect the methylation state of gene body.</span></p><ol start="2">
<li><span>Which states from DNA methylation segmentation (MethylSeekR) overlap with chromatin segmentation (ChromHMM)?</span><br>
<span>In the figure below, we can clearly see that for liver segmentation, state NS (no signal) is overlapping with HMR, Pr_A (active promoter) region is overlapping with UMR, Tx_W as well as Tx_S are overlapping with HMR. Kidney segmentation also follows the similar overlapping patterns.</span><br>
<img src="https://i.imgur.com/dVITP5d.png" alt="" loading="lazy"></li>
</ol><p><span>Note from the figure below, we observe that there is a state 14_Het_H3K9me3 (H3K9me3 heterochromatin) that overlaps with HMR in liver sample while in kidney sample 14_Het_H3K9me3 overlaps with PMD.</span><br>
<img src="https://i.imgur.com/dtrkY8v.png" alt="" loading="lazy"></p><p><span>Furthermore, there is a small region in the kidney sample with state NS that also overlaps with LMR.</span><br>
<img src="https://i.imgur.com/0p4qmUo.png" alt="" loading="lazy"></p><ol start="3">
<li><span>Do you observe overlap between the accessibility peaks and specific chromatin states? Which ones?</span><br>
<span>Yes, there are  accessibility peaks that overlaps with specific chromatin states. As we can see from the figure below, accessibility peaks are overlapping with Pr_A (active promoter) state.</span><br>
<img src="https://i.imgur.com/MsSIHiI.png" alt="" loading="lazy"></li>
</ol><p><span>Furthermore, we have a region of interest in the liver genome where accessibility peak is overlapping with Pr_B (bivalent promoter: Histone H3 lysine 4 monomethylation [H3K4me1] or trimethylation [H3K4me3] and repressing (e.g. H3K27me3)). So here we have different states of chromatin in liver and kidney genome that overlap with accessibility peaks.</span><br>
<img src="https://i.imgur.com/3683eJE.png" alt="" loading="lazy"></p><p><span>Besides from accessibility peak - overlapping - Pr_A, we encounter the genomic region in liver sample where the chromatin state is SEnh_pxS (Strong,TSS -prox Enhancer)	that overlaps with accessibility peak.</span><br>
<span>For comparison purpose, in kidney sample, peaks also appears that overlaps with Tx_W (weak transcription) but not as sharp as liver sample.</span><br>
<img src="https://i.imgur.com/gnDWwfx.png" alt="" loading="lazy"></p><p><span>So we can say that mostly active promoter regions are overlaps with accessibility peak because they are associated with chromtin accessibility. While there may also (may be few) exist other states than promoter regions that overlaps with accessibility peak.</span></p><p><span>CONSENSUS PEAKS: Peaks where both kindey and liver sample have chromatin accessibility.</span></p><ol start="4">
<li><span>Can you find some examples for unexpressed genes. What are the methylation states of their promoters and</span><br>
<span>bodies? Which histone marks are enriched at the promoters and bodies of these genes?</span></li>
</ol><p><span>Yes, there are some gene which are not expressed.</span><br>
<span>(a) Gene Gm3734: From the figure below, we see that the gene Gm3734 is not expressed as there are no signals in from RNAseq for both liver and kidney. Methylation states of its promoter and gene is PMD (partially methylated domain).</span><br>
<img src="https://i.imgur.com/SroK1Tj.png" alt="" loading="lazy"></p><p><span>(b) Gene Hdhd1a: We found another gene which is unexpressed. This gene (promoter as well as gene body) is lying in PMD. Furthermore, we can see for both liver and kidney sample, that histone marks H3k27me3 and H3k9me3 show higher signals on the gene body and also on promoter regions.</span><br>
<img src="https://i.imgur.com/fJL5upL.png" alt="" loading="lazy"></p><p><span>c) Gene Ftmt: Figure below explores an another instance of unexpressed gene with no RNAseq signals. Further, the promoter and gene body of the region are lying under HMR (for both kidney and liver samples) (from WGBS signals and segmentation). The histone marks that we observe in this region are H3K27me3 and H3K9me3 (from chipseq signals).</span><br>
<img src="https://i.imgur.com/caMHTOK.png" alt="" loading="lazy"></p><ol start="5">
<li><span>DNA &amp; chromatin accessibility = negatively associated</span><br>
<span>DNA &amp; gene expression = negatively associated</span></li>
</ol><p><img src="https://i.imgur.com/ph8hkfm.png" alt="" loading="lazy"></p><p><img src="https://i.imgur.com/t25CjFY.png" alt="" loading="lazy"></p><ol start="6">
<li><span>Do you observe positive or negative associations between DNA methylation, chromatin accessibility, individual histone marks and gene expression? Do the association differ in certain types of regions (e.g. promoters, gene bodies, enhancers)?</span><br>
<span>Negative association between DNA methylation and the marked histone marks (figure below).</span><br>
<img src="https://i.imgur.com/xMwob8b.png" alt="" loading="lazy"></li>
</ol><p><span>Negative association between DNA methylation and gene expression (figure below).</span><br>
<img src="https://i.imgur.com/p5KVLQ5.png" alt="" loading="lazy"></p><p><span>Positive association between DNA methylation (H3K4me3) and gene expression.</span><br>
<img src="https://i.imgur.com/Un0kfgr.png" alt="" loading="lazy"></p><h2 id="3-Integrative-analysis" data-id="3-Integrative-analysis"><a class="anchor hidden-xs" href="#3-Integrative-analysis" title="3-Integrative-analysis"><span class="octicon octicon-link"></span></a><span>3. Integrative analysis</span></h2><ol>
<li><span>Summarize signals over regions of interest</span><br>
<span>a) Define regions of interest (ROI) – generate BED3 files1 for one or more of the following regions:</span><br>
<span>ROI : promoter region (Pr)</span></li>
</ol><pre><code>grep "Pr" /vol/COMPEPIWS/groups/chipseq2/segmentation/relabelled/liver_relabelled_dense.bed | awk '{print $0}' &gt; /vol/COMPEPIWS/groups/chipseq2/analysis/Pmr.bed
    
computeMatrix scale-regions -S /vol/COMPEPIWS/groups/shared/ChIP-seq/chipseq2/signals/liver_14.5_H3K4me3_R1.bigWig /vol/COMPEPIWS/groups/shared/ChIP-seq/chipseq3/signals/kidney_14.5_H3K9me3_R1.bigWig -R Pmr.bed -o Pr_Comp_Enh
    
</code></pre><p><span>b) Combine computeMatrix with plotHeatmap/plotProfile from deepTools22 to generate plots. e.g. histone mark/accessibility/DNA methylation signals across all/some of the above mentioned ROI. What do you observe? Explain your findings from a gene-regulation perspective.</span></p><pre><code>plotProfile --matrixFile Pr_Comp_Enh --outFileName plotprofile_Pmr_CompEnh.png

plotHeatmap --matrixFile Pr_Comp_Enh --outFileName plotheatmap_Comp_Pmr_Enh.png
    
</code></pre><p><span>The figure below predicts that the histone mark H3K4me3 gives rise to gene expression while the histone mark H3K9me3 is not associated with gene expression. The black spots denotes the noise that should be rectified. Blue region denotes the gene expression, red region denote repressed region.</span><br>
<img src="https://i.imgur.com/dYw6ppi.png" alt="" loading="lazy"></p><ol start="2">
<li><span>Chromatin states and DNA methylation</span><br>
<span>a) Average DNA-methylation in the chromatin states (15-state model) of the corresponding samples</span></li>
</ol><pre><code>bedtools intersect -wa -wb -a "/vol/COMPEPIWS/groups/shared/WGBS/wgbs1/signal/rnbeads_sample_01_kidney.bedGraph" -b "/vol/COMPEPIWS/groups/shared/ChIP-seq/chipseq3/segmentation/kidney_relabeledstate15_dense.bed_dense.bed"    

bedtools groupby -i liver_avg_DNAme.bed -g 8 -c 4 -o mean &gt; liver_group.bed

In R:
ggplot(obj, aes(x=Histone, y=liver_me)) + geom_boxplot(fill="pink")
</code></pre><p><span>b)</span><br>
<img src="https://i.imgur.com/QxdWtJi.png" alt="" loading="lazy"><br>
<img src="https://i.imgur.com/HC5xD2E.png" alt="" loading="lazy"></p><p><span>c) From the figure we can see that the state Pr_A (active promoter) is low methylated in both kidney and liver with histone marks: H3K27ac, H3K9ac, H3K4me3. While transcription states are highly methylated with histone marks: H3K36me3.</span></p><p><span>d) Yes, there is a difference in the average methylation between two cell types. For instance, Pr_A is more methylated in kidney sample as compared to liver sample. Furthermore,Enh_dsS and Enh_pxS are more methylated in liver sample as compared to kidney sample.</span></p><ol start="4">
<li><span>Differentially Methylated Regions (DMRs) between kidney and liver: overlap with chromatin states and</span><br>
<span>accessibility</span><br>
<span>a)</span></li>
</ol><pre><code>cut -f 16-19 "/vol/COMPEPIWS/groups/shared/WGBS/wgbs1/differential/methdiff.tsv" |sed 's/"//g'  &gt; mythl.bed

grep Het "/vol/COMPEPIWS/groups/shared/ChIP-seq/chipseq2/segmentation/kidney_relabelled_dense.bed" &gt; kidney_DMR.bed

bedtools intersect -wa -wb -a mythl.bed -b kidney_DMR.bed | wc -l
</code></pre><p><span>528 and 1061 (for liver and kidney respectively).</span></p><p><img src="https://i.imgur.com/3mhiJgh.png" alt="" loading="lazy"></p></div>
    <div class="ui-toc dropup unselectable hidden-print" style="display:none;">
        <div class="pull-right dropdown">
            <a id="tocLabel" class="ui-toc-label btn btn-default" data-toggle="dropdown" href="#" role="button" aria-haspopup="true" aria-expanded="false" title="Table of content">
                <i class="fa fa-bars"></i>
            </a>
            <ul id="ui-toc" class="ui-toc-dropdown dropdown-menu" aria-labelledby="tocLabel">
                <div class="toc"><ul class="nav">
<li><a href="#The-environment---task-sheet" title="The environment - task sheet">The environment - task sheet</a><ul class="nav">
<li><a href="#2-Get-to-know-the-system" title="2 Get to know the system">2 Get to know the system</a></li>
<li><a href="#3-Tips-and-Tricks" title="3 Tips and Tricks">3 Tips and Tricks</a></li>
<li><a href="#4-Working-with-tables-on-the-command-line-or-awk" title="4 Working with tables on the command line, or awk">4 Working with tables on the command line, or awk</a></li>
<li><a href="#5-Conda-and-Bioconda" title="5 Conda and Bioconda">5 Conda and Bioconda</a></li>
<li><a href="#6-SLURM" title="6 SLURM">6 SLURM</a></li>
<li><a href="#7-Basics-in-R" title="7 Basics in R">7 Basics in R</a></li>
</ul>
</li>
<li><a href="#ChIP-seq-Assay" title="ChIP-seq Assay">ChIP-seq Assay</a><ul class="nav">
<li><a href="#Nextflow-ChIP-seq-pipeline" title="Nextflow ChIP-seq pipeline">Nextflow ChIP-seq pipeline</a><ul class="nav">
<li><a href="#1-Organize-your-raw-data" title="1. Organize your raw data">1. Organize your raw data</a></li>
<li><a href="#2-Generate-the-samplesheet" title="2. Generate the samplesheet">2. Generate the samplesheet</a></li>
<li><a href="#3-Running-the-pipeline" title="3. Running the pipeline">3. Running the pipeline</a></li>
</ul>
</li>
<li><a href="#2-Quality-Control" title="2 Quality Control">2 Quality Control</a></li>
<li><a href="#3-Exploratory-Analysis-using-IGV" title="3  Exploratory Analysis using IGV">3  Exploratory Analysis using IGV</a></li>
<li><a href="#4-Chromatin-segmentation-with-ChromHMM" title="4 Chromatin segmentation with ChromHMM">4 Chromatin segmentation with ChromHMM</a></li>
<li><a href="#5-Exploration-of-chromatin-state-segmentation-using-IGV" title="5 Exploration of chromatin state segmentation using IGV">5 Exploration of chromatin state segmentation using IGV</a></li>
<li><a href="#6-Exploratory-Analysis-using-R" title="6 Exploratory Analysis using R">6 Exploratory Analysis using R</a></li>
</ul>
</li>
<li><a href="#Integrative-analysis" title="Integrative analysis">Integrative analysis</a><ul class="nav">
<li><a href="#2-Integrative-data-exploration-in-IGV" title="2. Integrative data exploration in IGV">2. Integrative data exploration in IGV</a></li>
<li><a href="#3-Integrative-analysis" title="3. Integrative analysis">3. Integrative analysis</a></li>
</ul>
</li>
</ul>
</div><div class="toc-menu"><a class="expand-toggle" href="#">Expand all</a><a class="back-to-top" href="#">Back to top</a><a class="go-to-bottom" href="#">Go to bottom</a></div>
            </ul>
        </div>
    </div>
    <div id="ui-toc-affix" class="ui-affix-toc ui-toc-dropdown unselectable hidden-print" data-spy="affix" style="top:17px;display:none;" null null>
        <div class="toc"><ul class="nav">
<li><a href="#The-environment---task-sheet" title="The environment - task sheet">The environment - task sheet</a><ul class="nav">
<li><a href="#2-Get-to-know-the-system" title="2 Get to know the system">2 Get to know the system</a></li>
<li><a href="#3-Tips-and-Tricks" title="3 Tips and Tricks">3 Tips and Tricks</a></li>
<li><a href="#4-Working-with-tables-on-the-command-line-or-awk" title="4 Working with tables on the command line, or awk">4 Working with tables on the command line, or awk</a></li>
<li><a href="#5-Conda-and-Bioconda" title="5 Conda and Bioconda">5 Conda and Bioconda</a></li>
<li><a href="#6-SLURM" title="6 SLURM">6 SLURM</a></li>
<li><a href="#7-Basics-in-R" title="7 Basics in R">7 Basics in R</a></li>
</ul>
</li>
<li><a href="#ChIP-seq-Assay" title="ChIP-seq Assay">ChIP-seq Assay</a><ul class="nav">
<li><a href="#Nextflow-ChIP-seq-pipeline" title="Nextflow ChIP-seq pipeline">Nextflow ChIP-seq pipeline</a><ul class="nav">
<li><a href="#1-Organize-your-raw-data" title="1. Organize your raw data">1. Organize your raw data</a></li>
<li><a href="#2-Generate-the-samplesheet" title="2. Generate the samplesheet">2. Generate the samplesheet</a></li>
<li><a href="#3-Running-the-pipeline" title="3. Running the pipeline">3. Running the pipeline</a></li>
</ul>
</li>
<li><a href="#2-Quality-Control" title="2 Quality Control">2 Quality Control</a></li>
<li><a href="#3-Exploratory-Analysis-using-IGV" title="3  Exploratory Analysis using IGV">3  Exploratory Analysis using IGV</a></li>
<li><a href="#4-Chromatin-segmentation-with-ChromHMM" title="4 Chromatin segmentation with ChromHMM">4 Chromatin segmentation with ChromHMM</a></li>
<li><a href="#5-Exploration-of-chromatin-state-segmentation-using-IGV" title="5 Exploration of chromatin state segmentation using IGV">5 Exploration of chromatin state segmentation using IGV</a></li>
<li><a href="#6-Exploratory-Analysis-using-R" title="6 Exploratory Analysis using R">6 Exploratory Analysis using R</a></li>
</ul>
</li>
<li><a href="#Integrative-analysis" title="Integrative analysis">Integrative analysis</a><ul class="nav">
<li><a href="#2-Integrative-data-exploration-in-IGV" title="2. Integrative data exploration in IGV">2. Integrative data exploration in IGV</a></li>
<li><a href="#3-Integrative-analysis" title="3. Integrative analysis">3. Integrative analysis</a></li>
</ul>
</li>
</ul>
</div><div class="toc-menu"><a class="expand-toggle" href="#">Expand all</a><a class="back-to-top" href="#">Back to top</a><a class="go-to-bottom" href="#">Go to bottom</a></div>
    </div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js" integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous" defer></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gist-embed/2.6.0/gist-embed.min.js" integrity="sha256-KyF2D6xPIJUW5sUDSs93vWyZm+1RzIpKCexxElmxl8g=" crossorigin="anonymous" defer></script>
    <script>
        var markdown = $(".markdown-body");
        //smooth all hash trigger scrolling
        function smoothHashScroll() {
            var hashElements = $("a[href^='#']").toArray();
            for (var i = 0; i < hashElements.length; i++) {
                var element = hashElements[i];
                var $element = $(element);
                var hash = element.hash;
                if (hash) {
                    $element.on('click', function (e) {
                        // store hash
                        var hash = this.hash;
                        if ($(hash).length <= 0) return;
                        // prevent default anchor click behavior
                        e.preventDefault();
                        // animate
                        $('body, html').stop(true, true).animate({
                            scrollTop: $(hash).offset().top
                        }, 100, "linear", function () {
                            // when done, add hash to url
                            // (default click behaviour)
                            window.location.hash = hash;
                        });
                    });
                }
            }
        }

        smoothHashScroll();
        var toc = $('.ui-toc');
        var tocAffix = $('.ui-affix-toc');
        var tocDropdown = $('.ui-toc-dropdown');
        //toc
        tocDropdown.click(function (e) {
            e.stopPropagation();
        });

        var enoughForAffixToc = true;

        function generateScrollspy() {
            $(document.body).scrollspy({
                target: ''
            });
            $(document.body).scrollspy('refresh');
            if (enoughForAffixToc) {
                toc.hide();
                tocAffix.show();
            } else {
                tocAffix.hide();
                toc.show();
            }
            $(document.body).scroll();
        }

        function windowResize() {
            //toc right
            var paddingRight = parseFloat(markdown.css('padding-right'));
            var right = ($(window).width() - (markdown.offset().left + markdown.outerWidth() - paddingRight));
            toc.css('right', right + 'px');
            //affix toc left
            var newbool;
            var rightMargin = (markdown.parent().outerWidth() - markdown.outerWidth()) / 2;
            //for ipad or wider device
            if (rightMargin >= 133) {
                newbool = true;
                var affixLeftMargin = (tocAffix.outerWidth() - tocAffix.width()) / 2;
                var left = markdown.offset().left + markdown.outerWidth() - affixLeftMargin;
                tocAffix.css('left', left + 'px');
            } else {
                newbool = false;
            }
            if (newbool != enoughForAffixToc) {
                enoughForAffixToc = newbool;
                generateScrollspy();
            }
        }
        $(window).resize(function () {
            windowResize();
        });
        $(document).ready(function () {
            windowResize();
            generateScrollspy();
        });

        //remove hash
        function removeHash() {
            window.location.hash = '';
        }

        var backtotop = $('.back-to-top');
        var gotobottom = $('.go-to-bottom');

        backtotop.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            if (scrollToTop)
                scrollToTop();
            removeHash();
        });
        gotobottom.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            if (scrollToBottom)
                scrollToBottom();
            removeHash();
        });

        var toggle = $('.expand-toggle');
        var tocExpand = false;

        checkExpandToggle();
        toggle.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            tocExpand = !tocExpand;
            checkExpandToggle();
        })

        function checkExpandToggle () {
            var toc = $('.ui-toc-dropdown .toc');
            var toggle = $('.expand-toggle');
            if (!tocExpand) {
                toc.removeClass('expand');
                toggle.text('Expand all');
            } else {
                toc.addClass('expand');
                toggle.text('Collapse all');
            }
        }

        function scrollToTop() {
            $('body, html').stop(true, true).animate({
                scrollTop: 0
            }, 100, "linear");
        }

        function scrollToBottom() {
            $('body, html').stop(true, true).animate({
                scrollTop: $(document.body)[0].scrollHeight
            }, 100, "linear");
        }
    </script>
</body>

</html>
